{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1-Lifting SNP positions and merging the datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import norm\n",
    "from matplotlib_venn import venn3\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import subprocess\n",
    "import math\n",
    "import os\n",
    "import sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the path to Immuno and NeuroX files (BIM, BED and FAM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_path = \"data/IMMUNO/\"\n",
    "neurox_path = \"data/NEUROX/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explore the data contained in the two BIMs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_bim_fn = os.path.join(immuno_path, \"IMMUNO.bim\")\n",
    "neurox_bim_fn = os.path.join(neurox_path, \"NEUROX.bim\")\n",
    "\n",
    "immuno_bim = pd.read_csv(immuno_bim_fn, sep=\"\\t\", header=None)\n",
    "neurox_bim = pd.read_csv(neurox_bim_fn, sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>898835</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1108138</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1110294</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1125105</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1130298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1153667</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11260562</td>\n",
       "      <td>0</td>\n",
       "      <td>1155173</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6697886</td>\n",
       "      <td>0</td>\n",
       "      <td>1163474</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1_1168711</td>\n",
       "      <td>0</td>\n",
       "      <td>1168711</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6603785</td>\n",
       "      <td>0</td>\n",
       "      <td>1176365</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5\n",
       "0  1  imm_1_898835  0   898835  0  A\n",
       "1  1  vh_1_1108138  0  1108138  T  C\n",
       "2  1  vh_1_1110294  0  1110294  A  G\n",
       "3  1     rs9729550  0  1125105  C  A\n",
       "4  1     rs1815606  0  1130298  T  G\n",
       "5  1     rs7515488  0  1153667  T  C\n",
       "6  1    rs11260562  0  1155173  A  G\n",
       "7  1     rs6697886  0  1163474  A  G\n",
       "8  1     1_1168711  0  1168711  A  G\n",
       "9  1     rs6603785  0  1176365  T  A"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2       3  4  5\n",
       "0  1     NeuroX_PARK7_Pro158del  0       0  0  I\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I\n",
       "2  1   NeuroX_PINK1_534_535insQ  0       0  0  D\n",
       "3  1      NeuroX_PINK1_Asp525fs  0       0  0  D\n",
       "4  1      NeuroX_PINK1_Cys549fs  0       0  0  I\n",
       "5  1      NeuroX_PINK1_Lys520fs  0       0  0  I\n",
       "6  1                 exm2268640  0  762320  T  C\n",
       "7  1                      exm41  0  861349  0  C\n",
       "8  1                 exm1916089  0  865545  0  0\n",
       "9  1                      exm44  0  865584  0  G"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The SNPs contained in IMMUNO dataset have been mapped on hg18 genome assembly, while those in NEUROX have been mapped on hg19 assembly. To overcome this problem we must lift IMMUNO's SNPs positions to the corresponding hg19 positions. To accomplish this task we will use the web-based tool liftOver from UCSC (https://genome.ucsc.edu/cgi-bin/hgLiftOver)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since liftOver accepts as input UCSC BED files and not BIMs, we turn IMMUNO BIM in the corresponding UCSC BED file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      0        3        3             1\n",
      "0  chr1   898835   898836  imm_1_898835\n",
      "1  chr1  1108138  1108139  vh_1_1108138\n",
      "2  chr1  1110294  1110295  vh_1_1110294\n",
      "3  chr1  1125105  1125106     rs9729550\n",
      "4  chr1  1130298  1130299     rs1815606\n",
      "            0          3          3           1\n",
      "196519   chrX  154889906  154889907   rs3093525\n",
      "196520   chrX  154889917  154889918   rs3093526\n",
      "196521   chrX  154892630  154892631   rs2742301\n",
      "196522   chrX  154907376  154907377   rs2981835\n",
      "196523  chrMT       3721       3722  MitoA3721G\n"
     ]
    }
   ],
   "source": [
    "tmp_chr = immuno_bim.iloc[:,0]\n",
    "tmp_pos = immuno_bim.iloc[:,3]\n",
    "tmp_name = immuno_bim.iloc[:,1]\n",
    "\n",
    "# append 'chr' in fron of the chromosome number\n",
    "tmp_chr_str = [''.join([\"chr\", str(c)]) for c in tmp_chr.values]\n",
    "\n",
    "# replace chromosome numbers 23, 24, 25 and 26 with X, Y, X and MT\n",
    "tmp_chr_str2 = [c.replace('23', 'X') for c in tmp_chr_str]\n",
    "tmp_chr_str3 = [c.replace('24', 'Y') for c in tmp_chr_str2]\n",
    "tmp_chr_str4 = [c.replace('25', 'X') for c in tmp_chr_str3]\n",
    "tmp_chr_str5 = [c.replace('26', 'MT') for c in tmp_chr_str4]\n",
    "\n",
    "# write the resulting UCSC BED file\n",
    "bed = pd.concat(\n",
    "    [pd.DataFrame(tmp_chr_str5), tmp_pos, tmp_pos+1, tmp_name],\n",
    "    axis=1\n",
    ")\n",
    "print(bed.head(n=5))\n",
    "print(bed.tail(n=5))\n",
    "bed.to_csv(\n",
    "    os.path.join(immuno_path, \"IMMUNO_tolift.bed\"), sep=\"\\t\",\n",
    "    index=False, header=False\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LiftOver mapped succesfully 196,520 of IMMUNO SNPs, while for 4 of them the conversion failed. We renamed the resulting UCSC BED file (with the positions lifted) IMMUNO_lifted.bed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_lifted = pd.read_csv(os.path.join(immuno_path, \"IMMUNO_lifted.bed\"), sep=\"\\t\", header=None)\n",
    "\n",
    "# we take SNPs whose conversion to hg19 positions was succesful\n",
    "good_snps = bed_lifted.iloc[:,3]\n",
    "good_snps.to_csv(\n",
    "    os.path.join(immuno_path, \"good_snps.txt\"),\n",
    "    index=False, header=False\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As mentioned 4 SNPs were not converted to hg19 positions, thus we need to remove them from data in the BIM. To do this we use plink."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/IMMUNO/IMMUNO_hg19.log.\n",
      "Options in effect:\n",
      "  --bfile data/IMMUNO/IMMUNO\n",
      "  --extract data/IMMUNO/good_snps.txt\n",
      "  --make-bed\n",
      "  --out data/IMMUNO/IMMUNO_hg19\n",
      "  --update-map data/IMMUNO/IMMUNO_lifted.bed 2 4\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "196524 variants loaded from .bim file.\n",
      "523 people (343 males, 180 females) loaded from .fam.\n",
      "--update-map: 196520 values updated.\n",
      "Warning: Base-pair positions are now unsorted!\n",
      "--extract: 196520 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 523 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 67239 het. haploid genotypes present (see data/IMMUNO/IMMUNO_hg19.hh\n",
      "); many commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.957444.\n",
      "196520 variants and 523 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to data/IMMUNO/IMMUNO_hg19.bed + data/IMMUNO/IMMUNO_hg19.bim +\n",
      "data/IMMUNO/IMMUNO_hg19.fam ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, \"IMMUNO\")} --update-map {os.path.join(immuno_path, \"IMMUNO_lifted.bed\")} 2 4 --make-bed --extract {os.path.join(immuno_path, \"good_snps.txt\")} --out {os.path.join(immuno_path, \"IMMUNO_hg19\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, both IMMUNO and NEUROX datasets have SNPs mapped on hg19 genome assembly. Then, we can merge the two datasets in a single one, containing all the SNPs available in the two sets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1120431</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1163804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11260562</td>\n",
       "      <td>0</td>\n",
       "      <td>1165310</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6697886</td>\n",
       "      <td>0</td>\n",
       "      <td>1173611</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1_1168711</td>\n",
       "      <td>0</td>\n",
       "      <td>1178848</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6603785</td>\n",
       "      <td>0</td>\n",
       "      <td>1186502</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5\n",
       "0  1  imm_1_898835  0   908972  0  A\n",
       "1  1  vh_1_1108138  0  1118275  T  C\n",
       "2  1  vh_1_1110294  0  1120431  A  G\n",
       "3  1     rs9729550  0  1135242  C  A\n",
       "4  1     rs1815606  0  1140435  T  G\n",
       "5  1     rs7515488  0  1163804  T  C\n",
       "6  1    rs11260562  0  1165310  A  G\n",
       "7  1     rs6697886  0  1173611  A  G\n",
       "8  1     1_1168711  0  1178848  A  G\n",
       "9  1     rs6603785  0  1186502  T  A"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# consider the hg19 mapped BIM of IMMUNO\n",
    "immuno_bim = pd.read_csv(os.path.join(immuno_path, \"IMMUNO_hg19.bim\"), sep=\"\\t\", header=None)\n",
    "immuno_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2       3  4  5\n",
       "0  1     NeuroX_PARK7_Pro158del  0       0  0  I\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I\n",
       "2  1   NeuroX_PINK1_534_535insQ  0       0  0  D\n",
       "3  1      NeuroX_PINK1_Asp525fs  0       0  0  D\n",
       "4  1      NeuroX_PINK1_Cys549fs  0       0  0  I\n",
       "5  1      NeuroX_PINK1_Lys520fs  0       0  0  I\n",
       "6  1                 exm2268640  0  762320  T  C\n",
       "7  1                      exm41  0  861349  0  C\n",
       "8  1                 exm1916089  0  865545  0  0\n",
       "9  1                      exm44  0  865584  0  G"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before merging SNPs data, we must check how many subjects are in both NEUROX and IMMUNO datasets. Then we keep the data only for those whose data are in both the considered datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_fam_fn = os.path.join(immuno_path, \"IMMUNO_hg19.fam\")\n",
    "neurox_fam_fn = os.path.join(neurox_path, \"NEUROX.fam\")\n",
    "\n",
    "immuno_fam = pd.read_csv(immuno_fam_fn, sep=\" \", header=None)\n",
    "neurox_fam = pd.read_csv(neurox_fam_fn, sep=\" \", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3400</td>\n",
       "      <td>3400</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3401</td>\n",
       "      <td>3401</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3402</td>\n",
       "      <td>3402</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3403</td>\n",
       "      <td>3403</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3404</td>\n",
       "      <td>3404</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0     1  2  3  4  5\n",
       "0  3400  3400  0  0  2 -9\n",
       "1  3401  3401  0  0  2 -9\n",
       "2  3402  3402  0  0  1 -9\n",
       "3  3403  3403  0  0  1 -9\n",
       "4  3404  3404  0  0  2 -9"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_fam.head(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3527</td>\n",
       "      <td>3527</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3274</td>\n",
       "      <td>3274</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3220</td>\n",
       "      <td>3220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3467</td>\n",
       "      <td>3467</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3171</td>\n",
       "      <td>3171</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0     1  2  3  4  5\n",
       "0  3527  3527  0  0  1 -9\n",
       "1  3274  3274  0  0  1 -9\n",
       "2  3220  3220  0  0  2 -9\n",
       "3  3467  3467  0  0  1 -9\n",
       "4  3171  3171  0  0  1 -9"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_fam.head(n=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "immuno_subj = immuno_fam.iloc[:,1].tolist()\n",
    "neurox_subj = neurox_fam.iloc[:,1].tolist()\n",
    "\n",
    "# get the subjects appearing in both IMMUNO and NEUROX data\n",
    "common_subj = set(immuno_subj).intersection(neurox_subj)\n",
    "\n",
    "# write the common subjects to a file\n",
    "with open(\"data/common_subj.txt\", mode=\"w+\") as outfile:\n",
    "    for s in common_subj:\n",
    "        outfile.write(''.join([str(s), \" \", str(s), \"\\n\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In IMMUNO FAM file are contained 523 subjects, while in NEUROX FAM there are 619 subjects. The subjects whose genotyping data are available in both the considered dataset are 520. Then, before merging IMMUNO and NEUROX in a single dataset, we keep data belonging only to the subjects in common between the two datasets. To do this we use plink."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/IMMUNO/IMMUNO_common.log.\n",
      "Options in effect:\n",
      "  --bfile data/IMMUNO/IMMUNO_hg19\n",
      "  --keep data/common_subj.txt\n",
      "  --make-bed\n",
      "  --out data/IMMUNO/IMMUNO_common\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "196520 variants loaded from .bim file.\n",
      "523 people (343 males, 180 females) loaded from .fam.\n",
      "--keep: 520 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 66842 het. haploid genotypes present (see data/IMMUNO/IMMUNO_common.hh\n",
      "); many commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.957428.\n",
      "196520 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to data/IMMUNO/IMMUNO_common.bed + data/IMMUNO/IMMUNO_common.bim +\n",
      "data/IMMUNO/IMMUNO_common.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, \"IMMUNO_hg19\")} --keep data/common_subj.txt --make-bed --out {os.path.join(immuno_path, \"IMMUNO_common\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/NEUROX/NEUROX_common.log.\n",
      "Options in effect:\n",
      "  --bfile data/NEUROX/NEUROX\n",
      "  --keep data/common_subj.txt\n",
      "  --make-bed\n",
      "  --out data/NEUROX/NEUROX_common\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "267607 variants loaded from .bim file.\n",
      "619 people (409 males, 210 females) loaded from .fam.\n",
      "--keep: 520 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 4672 het. haploid genotypes present (see data/NEUROX/NEUROX_common.hh\n",
      "); many commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.957466.\n",
      "267607 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to data/NEUROX/NEUROX_common.bed + data/NEUROX/NEUROX_common.bim +\n",
      "data/NEUROX/NEUROX_common.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(neurox_path, \"NEUROX\")} --keep data/common_subj.txt --make-bed --out {os.path.join(neurox_path, \"NEUROX_common\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PPMI documentation tell us that the genotyping was made using custom genotype arrays. Thus, the SNP names do not always are identified through an rs ID. Moreover, we want to keep as much SNPs as possible from the two datasets: we need to keep the SNPs which are uniquely contained in IMMUNO and those uniquely contained in NEUROX and those shared by the two datasets. As mentioned before the SNP names are not always in rs ID format. Thus, to get the SNPs, which are uniquely in IMMUNO, we need to work on the genomic coordinates of the SNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "# the BIM are those containing the common subjects data\n",
    "immuno_bim = pd.read_csv(\n",
    "    os.path.join(immuno_path, \"IMMUNO_common.bim\"), sep=\"\\t\",\n",
    "    header=None\n",
    ")\n",
    "neurox_bim = pd.read_csv(\n",
    "    os.path.join(neurox_path, \"NEUROX_common.bim\"), sep=\"\\t\",\n",
    "    header=None\n",
    ")\n",
    "\n",
    "# identify each entry of the two datasets by its genomic coordinates\n",
    "chrs = immuno_bim.iloc[:,0]\n",
    "pos = immuno_bim.iloc[:,3]\n",
    "wt = immuno_bim.iloc[:,4]\n",
    "fa = immuno_bim.iloc[:,5]\n",
    "\n",
    "assert len(chrs) == len(pos)\n",
    "\n",
    "coords = list()\n",
    "for i in range(len(chrs)):\n",
    "    coord = \"_\".join([str(chrs[i]), str(pos[i]), str(pos[i] + 1), wt[i], fa[i]])\n",
    "    coords.append(coord)\n",
    "immuno_coords = immuno_bim\n",
    "immuno_coords[6] = coords\n",
    "\n",
    "chrs = neurox_bim.iloc[:,0]\n",
    "pos = neurox_bim.iloc[:,3]\n",
    "wt = neurox_bim.iloc[:,4]\n",
    "fa = neurox_bim.iloc[:,5]\n",
    "\n",
    "assert len(chrs) == len(pos)\n",
    "\n",
    "coords = list()\n",
    "for i in range(len(chrs)):\n",
    "    coord = \"_\".join([str(chrs[i]), str(pos[i]), str(pos[i] + 1), wt[i], fa[i]])\n",
    "    coords.append(coord)\n",
    "neurox_coords = neurox_bim\n",
    "neurox_coords[6] = coords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>1_908972_908973_0_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1118275_1118276_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1120431</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1120431_1120432_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1_1135242_1135243_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1140435_1140436_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1163804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1163804_1163805_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11260562</td>\n",
       "      <td>0</td>\n",
       "      <td>1165310</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1165310_1165311_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6697886</td>\n",
       "      <td>0</td>\n",
       "      <td>1173611</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1173611_1173612_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1_1168711</td>\n",
       "      <td>0</td>\n",
       "      <td>1178848</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1178848_1178849_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6603785</td>\n",
       "      <td>0</td>\n",
       "      <td>1186502</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1_1186502_1186503_T_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5                      6\n",
       "0  1  imm_1_898835  0   908972  0  A    1_908972_908973_0_A\n",
       "1  1  vh_1_1108138  0  1118275  T  C  1_1118275_1118276_T_C\n",
       "2  1  vh_1_1110294  0  1120431  A  G  1_1120431_1120432_A_G\n",
       "3  1     rs9729550  0  1135242  C  A  1_1135242_1135243_C_A\n",
       "4  1     rs1815606  0  1140435  T  G  1_1140435_1140436_T_G\n",
       "5  1     rs7515488  0  1163804  T  C  1_1163804_1163805_T_C\n",
       "6  1    rs11260562  0  1165310  A  G  1_1165310_1165311_A_G\n",
       "7  1     rs6697886  0  1173611  A  G  1_1173611_1173612_A_G\n",
       "8  1     1_1168711  0  1178848  A  G  1_1178848_1178849_A_G\n",
       "9  1     rs6603785  0  1186502  T  A  1_1186502_1186503_T_A"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_coords.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1_0_I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1_0_I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>1_0_1_0_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>1_0_1_0_D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1_0_I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1_0_I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_762320_762321_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>1_861349_861350_0_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1_865545_865546_0_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865584_865585_0_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2       3  4  5                    6\n",
       "0  1     NeuroX_PARK7_Pro158del  0       0  0  I            1_0_1_0_I\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I            1_0_1_0_I\n",
       "2  1   NeuroX_PINK1_534_535insQ  0       0  0  D            1_0_1_0_D\n",
       "3  1      NeuroX_PINK1_Asp525fs  0       0  0  D            1_0_1_0_D\n",
       "4  1      NeuroX_PINK1_Cys549fs  0       0  0  I            1_0_1_0_I\n",
       "5  1      NeuroX_PINK1_Lys520fs  0       0  0  I            1_0_1_0_I\n",
       "6  1                 exm2268640  0  762320  T  C  1_762320_762321_T_C\n",
       "7  1                      exm41  0  861349  0  C  1_861349_861350_0_C\n",
       "8  1                 exm1916089  0  865545  0  0  1_865545_865546_0_0\n",
       "9  1                      exm44  0  865584  0  G  1_865584_865585_0_G"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_coords.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# we get the names of the SNPs only available in IMMUNO dataset\n",
    "neurox_coords_set = set(neurox_coords.iloc[:,6].to_list())\n",
    "immuno_coords_list = immuno_coords.iloc[:,6].tolist()\n",
    "immuno_snp_names = immuno_coords.iloc[:,1].tolist()\n",
    "immuno_unique_snps = list()\n",
    "for i in range(len(immuno_coords_list)):\n",
    "    c = immuno_coords_list[i]\n",
    "    if c not in neurox_coords_set:\n",
    "        immuno_unique_snps.append(immuno_snp_names[i])\n",
    "        \n",
    "# write the SNPs uniquely appearing in IMMUNO\n",
    "with open(\"data/onlyIMMUNOsnps.txt\", mode=\"w+\") as outfile:\n",
    "    for snp in immuno_unique_snps:\n",
    "        outfile.write(''.join([snp, \"\\n\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then from IMMUNO dataset we keep only data related to the SNPs uniquely found in IMMUNO and not in NEUROX dataset. To accomlish this task, we use plink --extract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/IMMUNO/IMMUNO_uniquesnps.log.\n",
      "Options in effect:\n",
      "  --bfile data/IMMUNO/IMMUNO_common\n",
      "  --extract data/onlyIMMUNOsnps.txt\n",
      "  --make-bed\n",
      "  --out data/IMMUNO/IMMUNO_uniquesnps\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "196520 variants loaded from .bim file.\n",
      "520 people (341 males, 179 females) loaded from .fam.\n",
      "--extract: 189564 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 66671 het. haploid genotypes present (see\n",
      "data/IMMUNO/IMMUNO_uniquesnps.hh ); many commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.956414.\n",
      "189564 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to data/IMMUNO/IMMUNO_uniquesnps.bed +\n",
      "data/IMMUNO/IMMUNO_uniquesnps.bim + data/IMMUNO/IMMUNO_uniquesnps.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, \"IMMUNO_common\")} --extract data/onlyIMMUNOsnps.txt --make-bed --out {os.path.join(immuno_path, \"IMMUNO_uniquesnps\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can merge NEUROX and IMMUNO datasets in a single plink file containing the SNPs of both the considered datsets. To do this we use agian plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/PPMI_merged.log.\n",
      "Options in effect:\n",
      "  --bfile data/NEUROX/NEUROX_common\n",
      "  --bmerge data/IMMUNO/IMMUNO_uniquesnps\n",
      "  --make-bed\n",
      "  --out data/PPMI_merged\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "520 people loaded from data/NEUROX/NEUROX_common.fam.\n",
      "520 people to be merged from data/IMMUNO/IMMUNO_uniquesnps.fam.\n",
      "Of these, 0 are new, while 520 are present in the base dataset.\n",
      "267607 markers loaded from data/NEUROX/NEUROX_common.bim.\n",
      "189564 markers to be merged from data/IMMUNO/IMMUNO_uniquesnps.bim.\n",
      "Of these, 189564 are new, while 0 are present in the base dataset.\n",
      "Warning: Variants 'NeuroX_PINK1_23bp_del_ex7' and 'NeuroX_PARK7_Pro158del' have\n",
      "the same position.\n",
      "Warning: Variants 'NeuroX_PINK1_534_535insQ' and 'NeuroX_PINK1_23bp_del_ex7'\n",
      "have the same position.\n",
      "Warning: Variants 'NeuroX_PINK1_Asp525fs' and 'NeuroX_PINK1_534_535insQ' have\n",
      "the same position.\n",
      "3325 more same-position warnings: see log file.\n",
      "Performing single-pass merge (520 people, 457171 variants).\n",
      "Merged fileset written to data/PPMI_merged-merge.bed +\n",
      "data/PPMI_merged-merge.bim + data/PPMI_merged-merge.fam .\n",
      "457171 variants loaded from .bim file.\n",
      "520 people (341 males, 179 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 71343 het. haploid genotypes present (see data/PPMI_merged.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.95703.\n",
      "457171 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to data/PPMI_merged.bed + data/PPMI_merged.bim +\n",
      "data/PPMI_merged.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(neurox_path, \"NEUROX_common\")} --bmerge {os.path.join(immuno_path, \"IMMUNO_uniquesnps\")} --make-bed --out data/PPMI_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
