{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, sys\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import math\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from qqman import qqman"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATNO</th>\n",
       "      <th>CAUDATE_R</th>\n",
       "      <th>CAUDATE_L</th>\n",
       "      <th>PUTAMEN_R</th>\n",
       "      <th>PUTAMEN_L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>2.99</td>\n",
       "      <td>3.43</td>\n",
       "      <td>2.328620</td>\n",
       "      <td>1.973532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>1.56</td>\n",
       "      <td>1.12</td>\n",
       "      <td>-0.246956</td>\n",
       "      <td>-0.796382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>2.13</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.255480</td>\n",
       "      <td>0.339528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3003</td>\n",
       "      <td>2.19</td>\n",
       "      <td>1.27</td>\n",
       "      <td>-0.118658</td>\n",
       "      <td>-0.401457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3004</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.30</td>\n",
       "      <td>2.949289</td>\n",
       "      <td>2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>75484</td>\n",
       "      <td>1.35</td>\n",
       "      <td>1.24</td>\n",
       "      <td>-0.765017</td>\n",
       "      <td>-1.158726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>75505</td>\n",
       "      <td>1.90</td>\n",
       "      <td>1.66</td>\n",
       "      <td>0.065569</td>\n",
       "      <td>-0.486548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>75524</td>\n",
       "      <td>1.36</td>\n",
       "      <td>1.76</td>\n",
       "      <td>-0.665445</td>\n",
       "      <td>0.269728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>75537</td>\n",
       "      <td>1.46</td>\n",
       "      <td>1.03</td>\n",
       "      <td>0.065569</td>\n",
       "      <td>-1.070600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>75562</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.10</td>\n",
       "      <td>-1.773705</td>\n",
       "      <td>-1.573294</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>995 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PATNO  CAUDATE_R  CAUDATE_L  PUTAMEN_R  PUTAMEN_L\n",
       "0     3000       2.99       3.43   2.328620   1.973532\n",
       "1     3001       1.56       1.12  -0.246956  -0.796382\n",
       "2     3002       2.13       2.28   0.255480   0.339528\n",
       "3     3003       2.19       1.27  -0.118658  -0.401457\n",
       "4     3004       5.09       5.30   2.949289   2.400680\n",
       "..     ...        ...        ...        ...        ...\n",
       "990  75484       1.35       1.24  -0.765017  -1.158726\n",
       "991  75505       1.90       1.66   0.065569  -0.486548\n",
       "992  75524       1.36       1.76  -0.665445   0.269728\n",
       "993  75537       1.46       1.03   0.065569  -1.070600\n",
       "994  75562       1.00       1.10  -1.773705  -1.573294\n",
       "\n",
       "[995 rows x 5 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DATScan = pd.read_csv(\"../Imaging/DATScan_Norm.csv\")\n",
    "images = DATScan.iloc[:,np.r_[0,3:7]]\n",
    "images = images.groupby('PATNO').first().reset_index()\n",
    "images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>REFERRAL</th>\n",
       "      <th>DECLINED</th>\n",
       "      <th>RSNDEC</th>\n",
       "      <th>EXCLUDED</th>\n",
       "      <th>RSNEXC</th>\n",
       "      <th>ORIG_ENTRY</th>\n",
       "      <th>LAST_UPDATE</th>\n",
       "      <th>QUERY</th>\n",
       "      <th>SITE_APRV</th>\n",
       "      <th>APPRDX2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>0.021562</td>\n",
       "      <td>-0.005305</td>\n",
       "      <td>0.012287</td>\n",
       "      <td>0.065969</td>\n",
       "      <td>0.006345</td>\n",
       "      <td>0.011038</td>\n",
       "      <td>-0.051736</td>\n",
       "      <td>0.020446</td>\n",
       "      <td>...</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>2011-01-19 12:44:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>0.013209</td>\n",
       "      <td>0.016830</td>\n",
       "      <td>0.014295</td>\n",
       "      <td>-0.033689</td>\n",
       "      <td>-0.039806</td>\n",
       "      <td>0.047147</td>\n",
       "      <td>-0.026051</td>\n",
       "      <td>0.040981</td>\n",
       "      <td>...</td>\n",
       "      <td>60.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2011</td>\n",
       "      <td>2011-02-10 10:43:30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>-0.082064</td>\n",
       "      <td>-0.030599</td>\n",
       "      <td>-0.021643</td>\n",
       "      <td>-0.001498</td>\n",
       "      <td>0.025769</td>\n",
       "      <td>-0.005442</td>\n",
       "      <td>0.006985</td>\n",
       "      <td>-0.015841</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-04 11:04:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>0.023291</td>\n",
       "      <td>0.022143</td>\n",
       "      <td>-0.013579</td>\n",
       "      <td>-0.013314</td>\n",
       "      <td>-0.029992</td>\n",
       "      <td>-0.025549</td>\n",
       "      <td>-0.115247</td>\n",
       "      <td>-0.001666</td>\n",
       "      <td>...</td>\n",
       "      <td>99.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-30 11:43:13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>0.008937</td>\n",
       "      <td>0.057132</td>\n",
       "      <td>0.088903</td>\n",
       "      <td>0.031578</td>\n",
       "      <td>0.002286</td>\n",
       "      <td>0.036026</td>\n",
       "      <td>-0.031903</td>\n",
       "      <td>-0.001299</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-31 07:21:03.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>0.007515</td>\n",
       "      <td>0.030850</td>\n",
       "      <td>0.032300</td>\n",
       "      <td>0.006790</td>\n",
       "      <td>0.075679</td>\n",
       "      <td>-0.048470</td>\n",
       "      <td>0.028239</td>\n",
       "      <td>-0.062672</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-02-22 12:57:46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>-0.015873</td>\n",
       "      <td>0.026348</td>\n",
       "      <td>-0.005282</td>\n",
       "      <td>-0.022791</td>\n",
       "      <td>-0.076218</td>\n",
       "      <td>-0.045724</td>\n",
       "      <td>-0.004484</td>\n",
       "      <td>0.017099</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-12-10 12:36:00.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>05/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>-0.006236</td>\n",
       "      <td>-0.004422</td>\n",
       "      <td>-0.054059</td>\n",
       "      <td>0.034937</td>\n",
       "      <td>0.011776</td>\n",
       "      <td>-0.098040</td>\n",
       "      <td>-0.040385</td>\n",
       "      <td>0.032559</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-27 08:35:42.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.036144</td>\n",
       "      <td>-0.015858</td>\n",
       "      <td>-0.064136</td>\n",
       "      <td>-0.033492</td>\n",
       "      <td>0.042224</td>\n",
       "      <td>-0.105760</td>\n",
       "      <td>0.021767</td>\n",
       "      <td>-0.099659</td>\n",
       "      <td>...</td>\n",
       "      <td>99.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-06 07:09:02.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>0.038298</td>\n",
       "      <td>-0.099336</td>\n",
       "      <td>-0.061020</td>\n",
       "      <td>-0.021126</td>\n",
       "      <td>0.006314</td>\n",
       "      <td>-0.024565</td>\n",
       "      <td>0.026351</td>\n",
       "      <td>-0.048278</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-05-16 13:26:16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>HC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>480 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID       PC1       PC2       PC3       PC4       PC5       PC6  \\\n",
       "0    3000  3000  0.021562 -0.005305  0.012287  0.065969  0.006345  0.011038   \n",
       "1    3001  3001  0.013209  0.016830  0.014295 -0.033689 -0.039806  0.047147   \n",
       "2    3002  3002 -0.082064 -0.030599 -0.021643 -0.001498  0.025769 -0.005442   \n",
       "3    3004  3004  0.023291  0.022143 -0.013579 -0.013314 -0.029992 -0.025549   \n",
       "4    3006  3006  0.008937  0.057132  0.088903  0.031578  0.002286  0.036026   \n",
       "..    ...   ...       ...       ...       ...       ...       ...       ...   \n",
       "475  4124  4124  0.007515  0.030850  0.032300  0.006790  0.075679 -0.048470   \n",
       "476  4125  4125 -0.015873  0.026348 -0.005282 -0.022791 -0.076218 -0.045724   \n",
       "477  4126  4126 -0.006236 -0.004422 -0.054059  0.034937  0.011776 -0.098040   \n",
       "478  4136  4136  0.036144 -0.015858 -0.064136 -0.033492  0.042224 -0.105760   \n",
       "479  4139  4139  0.038298 -0.099336 -0.061020 -0.021126  0.006314 -0.024565   \n",
       "\n",
       "          PC7       PC8  ...  REFERRAL  DECLINED  RSNDEC EXCLUDED  RSNEXC  \\\n",
       "0   -0.051736  0.020446  ...      31.0       NaN     NaN      NaN     NaN   \n",
       "1   -0.026051  0.040981  ...      60.0       NaN     NaN      NaN     NaN   \n",
       "2    0.006985 -0.015841  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "3   -0.115247 -0.001666  ...      99.0       NaN     NaN      NaN     NaN   \n",
       "4   -0.031903 -0.001299  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "..        ...       ...  ...       ...       ...     ...      ...     ...   \n",
       "475  0.028239 -0.062672  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "476 -0.004484  0.017099  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "477 -0.040385  0.032559  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "478  0.021767 -0.099659  ...      99.0       NaN     NaN      NaN     NaN   \n",
       "479  0.026351 -0.048278  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "\n",
       "    ORIG_ENTRY            LAST_UPDATE  QUERY SITE_APRV  APPRDX2  \n",
       "0      01/2011  2011-01-19 12:44:58.0    NaN   01/2011       HC  \n",
       "1      02/2011  2011-02-10 10:43:30.0    NaN   03/2011       PD  \n",
       "2      03/2011  2011-03-04 11:04:58.0    NaN   03/2011       PD  \n",
       "3      03/2011  2011-03-30 11:43:13.0    NaN   03/2011       HC  \n",
       "4      03/2011  2011-03-31 07:21:03.0    NaN   03/2011       PD  \n",
       "..         ...                    ...    ...       ...      ...  \n",
       "475    02/2013  2013-02-22 12:57:46.0    NaN   02/2013       PD  \n",
       "476    02/2013  2013-12-10 12:36:00.0    NaN   05/2013       PD  \n",
       "477    03/2013  2013-03-27 08:35:42.0    NaN   03/2013       PD  \n",
       "478    03/2013  2013-03-06 07:09:02.0    NaN   06/2013       PD  \n",
       "479    03/2013  2013-05-16 13:26:16.0    NaN   06/2013       HC  \n",
       "\n",
       "[480 rows x 42 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppmi_info = pd.read_csv('../PPMI_data/Info files/Screening___Demographics.csv')\n",
    "\n",
    "pca = pd.read_table('Data/QC/White.eigenvec', sep=' ', header=None)\n",
    "cnames = ['FID','IID']\n",
    "for i in range(1,11):\n",
    "    cnames.append(\"PC\"+str(i))\n",
    "pca.columns = cnames\n",
    "\n",
    "info_pca = pca.merge(ppmi_info, left_on=['FID'], right_on=['PATNO'])\n",
    "\n",
    "apprdx = list()\n",
    "\n",
    "for i in info_pca['APPRDX'].to_list():\n",
    "    if(i==1):\n",
    "        apprdx.append(\"PD\")\n",
    "    if(i==2):\n",
    "        apprdx.append(\"HC\")\n",
    "    elif(i==3):\n",
    "        apprdx.append(\"SWEDD\")\n",
    "\n",
    "info_pca['APPRDX2'] = apprdx\n",
    "info_pca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>ORIG_ENTRY</th>\n",
       "      <th>LAST_UPDATE</th>\n",
       "      <th>QUERY</th>\n",
       "      <th>SITE_APRV</th>\n",
       "      <th>APPRDX2</th>\n",
       "      <th>PATNO_y</th>\n",
       "      <th>CAUDATE_R</th>\n",
       "      <th>CAUDATE_L</th>\n",
       "      <th>PUTAMEN_R</th>\n",
       "      <th>PUTAMEN_L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>0.021562</td>\n",
       "      <td>-0.005305</td>\n",
       "      <td>0.012287</td>\n",
       "      <td>0.065969</td>\n",
       "      <td>0.006345</td>\n",
       "      <td>0.011038</td>\n",
       "      <td>-0.051736</td>\n",
       "      <td>0.020446</td>\n",
       "      <td>...</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>2011-01-19 12:44:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>HC</td>\n",
       "      <td>3000</td>\n",
       "      <td>2.99</td>\n",
       "      <td>3.43</td>\n",
       "      <td>2.328620</td>\n",
       "      <td>1.973532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>0.013209</td>\n",
       "      <td>0.016830</td>\n",
       "      <td>0.014295</td>\n",
       "      <td>-0.033689</td>\n",
       "      <td>-0.039806</td>\n",
       "      <td>0.047147</td>\n",
       "      <td>-0.026051</td>\n",
       "      <td>0.040981</td>\n",
       "      <td>...</td>\n",
       "      <td>02/2011</td>\n",
       "      <td>2011-02-10 10:43:30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "      <td>3001</td>\n",
       "      <td>1.56</td>\n",
       "      <td>1.12</td>\n",
       "      <td>-0.246956</td>\n",
       "      <td>-0.796382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>-0.082064</td>\n",
       "      <td>-0.030599</td>\n",
       "      <td>-0.021643</td>\n",
       "      <td>-0.001498</td>\n",
       "      <td>0.025769</td>\n",
       "      <td>-0.005442</td>\n",
       "      <td>0.006985</td>\n",
       "      <td>-0.015841</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-04 11:04:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "      <td>3002</td>\n",
       "      <td>2.13</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.255480</td>\n",
       "      <td>0.339528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>0.023291</td>\n",
       "      <td>0.022143</td>\n",
       "      <td>-0.013579</td>\n",
       "      <td>-0.013314</td>\n",
       "      <td>-0.029992</td>\n",
       "      <td>-0.025549</td>\n",
       "      <td>-0.115247</td>\n",
       "      <td>-0.001666</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-30 11:43:13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>HC</td>\n",
       "      <td>3004</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.30</td>\n",
       "      <td>2.949289</td>\n",
       "      <td>2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>0.008937</td>\n",
       "      <td>0.057132</td>\n",
       "      <td>0.088903</td>\n",
       "      <td>0.031578</td>\n",
       "      <td>0.002286</td>\n",
       "      <td>0.036026</td>\n",
       "      <td>-0.031903</td>\n",
       "      <td>-0.001299</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-31 07:21:03.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "      <td>3006</td>\n",
       "      <td>2.28</td>\n",
       "      <td>2.12</td>\n",
       "      <td>0.175997</td>\n",
       "      <td>-2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>0.007515</td>\n",
       "      <td>0.030850</td>\n",
       "      <td>0.032300</td>\n",
       "      <td>0.006790</td>\n",
       "      <td>0.075679</td>\n",
       "      <td>-0.048470</td>\n",
       "      <td>0.028239</td>\n",
       "      <td>-0.062672</td>\n",
       "      <td>...</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-02-22 12:57:46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4124</td>\n",
       "      <td>1.95</td>\n",
       "      <td>1.59</td>\n",
       "      <td>-0.273727</td>\n",
       "      <td>-0.053985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>-0.015873</td>\n",
       "      <td>0.026348</td>\n",
       "      <td>-0.005282</td>\n",
       "      <td>-0.022791</td>\n",
       "      <td>-0.076218</td>\n",
       "      <td>-0.045724</td>\n",
       "      <td>-0.004484</td>\n",
       "      <td>0.017099</td>\n",
       "      <td>...</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-12-10 12:36:00.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>05/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4125</td>\n",
       "      <td>2.03</td>\n",
       "      <td>2.47</td>\n",
       "      <td>0.391926</td>\n",
       "      <td>1.005487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>-0.006236</td>\n",
       "      <td>-0.004422</td>\n",
       "      <td>-0.054059</td>\n",
       "      <td>0.034937</td>\n",
       "      <td>0.011776</td>\n",
       "      <td>-0.098040</td>\n",
       "      <td>-0.040385</td>\n",
       "      <td>0.032559</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-27 08:35:42.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4126</td>\n",
       "      <td>1.54</td>\n",
       "      <td>1.88</td>\n",
       "      <td>-0.219232</td>\n",
       "      <td>-1.158726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.036144</td>\n",
       "      <td>-0.015858</td>\n",
       "      <td>-0.064136</td>\n",
       "      <td>-0.033492</td>\n",
       "      <td>0.042224</td>\n",
       "      <td>-0.105760</td>\n",
       "      <td>0.021767</td>\n",
       "      <td>-0.099659</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-06 07:09:02.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.57</td>\n",
       "      <td>-1.406477</td>\n",
       "      <td>-1.922162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>0.038298</td>\n",
       "      <td>-0.099336</td>\n",
       "      <td>-0.061020</td>\n",
       "      <td>-0.021126</td>\n",
       "      <td>0.006314</td>\n",
       "      <td>-0.024565</td>\n",
       "      <td>0.026351</td>\n",
       "      <td>-0.048278</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-05-16 13:26:16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>HC</td>\n",
       "      <td>4139</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.82</td>\n",
       "      <td>1.349565</td>\n",
       "      <td>1.066702</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>475 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID       PC1       PC2       PC3       PC4       PC5       PC6  \\\n",
       "0    3000  3000  0.021562 -0.005305  0.012287  0.065969  0.006345  0.011038   \n",
       "1    3001  3001  0.013209  0.016830  0.014295 -0.033689 -0.039806  0.047147   \n",
       "2    3002  3002 -0.082064 -0.030599 -0.021643 -0.001498  0.025769 -0.005442   \n",
       "3    3004  3004  0.023291  0.022143 -0.013579 -0.013314 -0.029992 -0.025549   \n",
       "4    3006  3006  0.008937  0.057132  0.088903  0.031578  0.002286  0.036026   \n",
       "..    ...   ...       ...       ...       ...       ...       ...       ...   \n",
       "470  4124  4124  0.007515  0.030850  0.032300  0.006790  0.075679 -0.048470   \n",
       "471  4125  4125 -0.015873  0.026348 -0.005282 -0.022791 -0.076218 -0.045724   \n",
       "472  4126  4126 -0.006236 -0.004422 -0.054059  0.034937  0.011776 -0.098040   \n",
       "473  4136  4136  0.036144 -0.015858 -0.064136 -0.033492  0.042224 -0.105760   \n",
       "474  4139  4139  0.038298 -0.099336 -0.061020 -0.021126  0.006314 -0.024565   \n",
       "\n",
       "          PC7       PC8  ...  ORIG_ENTRY            LAST_UPDATE  QUERY  \\\n",
       "0   -0.051736  0.020446  ...     01/2011  2011-01-19 12:44:58.0    NaN   \n",
       "1   -0.026051  0.040981  ...     02/2011  2011-02-10 10:43:30.0    NaN   \n",
       "2    0.006985 -0.015841  ...     03/2011  2011-03-04 11:04:58.0    NaN   \n",
       "3   -0.115247 -0.001666  ...     03/2011  2011-03-30 11:43:13.0    NaN   \n",
       "4   -0.031903 -0.001299  ...     03/2011  2011-03-31 07:21:03.0    NaN   \n",
       "..        ...       ...  ...         ...                    ...    ...   \n",
       "470  0.028239 -0.062672  ...     02/2013  2013-02-22 12:57:46.0    NaN   \n",
       "471 -0.004484  0.017099  ...     02/2013  2013-12-10 12:36:00.0    NaN   \n",
       "472 -0.040385  0.032559  ...     03/2013  2013-03-27 08:35:42.0    NaN   \n",
       "473  0.021767 -0.099659  ...     03/2013  2013-03-06 07:09:02.0    NaN   \n",
       "474  0.026351 -0.048278  ...     03/2013  2013-05-16 13:26:16.0    NaN   \n",
       "\n",
       "    SITE_APRV  APPRDX2 PATNO_y CAUDATE_R  CAUDATE_L PUTAMEN_R  PUTAMEN_L  \n",
       "0     01/2011       HC    3000      2.99       3.43  2.328620   1.973532  \n",
       "1     03/2011       PD    3001      1.56       1.12 -0.246956  -0.796382  \n",
       "2     03/2011       PD    3002      2.13       2.28  0.255480   0.339528  \n",
       "3     03/2011       HC    3004      5.09       5.30  2.949289   2.400680  \n",
       "4     03/2011       PD    3006      2.28       2.12  0.175997  -2.400680  \n",
       "..        ...      ...     ...       ...        ...       ...        ...  \n",
       "470   02/2013       PD    4124      1.95       1.59 -0.273727  -0.053985  \n",
       "471   05/2013       PD    4125      2.03       2.47  0.391926   1.005487  \n",
       "472   03/2013       PD    4126      1.54       1.88 -0.219232  -1.158726  \n",
       "473   06/2013       PD    4136      0.68       0.57 -1.406477  -1.922162  \n",
       "474   06/2013       HC    4139      2.44       2.82  1.349565   1.066702  \n",
       "\n",
       "[475 rows x 47 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_pheno = info_pca.merge(images, left_on=['FID'], right_on=['PATNO'])\n",
    "info_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "info_pheno.iloc[:,0:2].to_csv(\"Data/QC/IDs_White.txt\", header=False, index=False, sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to Data/QC/White_2.log.\n",
      "Options in effect:\n",
      "  --bfile Data/QC/White\n",
      "  --keep Data/QC/IDs_White.txt\n",
      "  --make-bed\n",
      "  --out Data/QC/White_2\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "480 people (317 males, 163 females) loaded from .fam.\n",
      "--keep: 475 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to Data/QC/White_2.bed + Data/QC/White_2.bim + Data/QC/White_2.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile Data/QC/White --keep Data/QC/IDs_White.txt --make-bed --out Data/QC/White_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to Data/QC/White_2.log.\n",
      "Options in effect:\n",
      "  --bfile Data/QC/White_2\n",
      "  --out Data/QC/White_2\n",
      "  --pca 10\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "Using up to 11 threads (change this with --threads).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to Data/QC/White_2.eigenval and Data/QC/White_2.eigenvec .\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile Data/QC/White_2 --pca 10 --out Data/QC/White_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>APPRDX2</th>\n",
       "      <th>CAUDATE_R</th>\n",
       "      <th>CAUDATE_L</th>\n",
       "      <th>PUTAMEN_R</th>\n",
       "      <th>PUTAMEN_L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.99</td>\n",
       "      <td>3.43</td>\n",
       "      <td>2.328620</td>\n",
       "      <td>1.973532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.56</td>\n",
       "      <td>1.12</td>\n",
       "      <td>-0.246956</td>\n",
       "      <td>-0.796382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>PD</td>\n",
       "      <td>2.13</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.255480</td>\n",
       "      <td>0.339528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>HC</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.30</td>\n",
       "      <td>2.949289</td>\n",
       "      <td>2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>PD</td>\n",
       "      <td>2.28</td>\n",
       "      <td>2.12</td>\n",
       "      <td>0.175997</td>\n",
       "      <td>-2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.95</td>\n",
       "      <td>1.59</td>\n",
       "      <td>-0.273727</td>\n",
       "      <td>-0.053985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>PD</td>\n",
       "      <td>2.03</td>\n",
       "      <td>2.47</td>\n",
       "      <td>0.391926</td>\n",
       "      <td>1.005487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.54</td>\n",
       "      <td>1.88</td>\n",
       "      <td>-0.219232</td>\n",
       "      <td>-1.158726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>PD</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.57</td>\n",
       "      <td>-1.406477</td>\n",
       "      <td>-1.922162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.82</td>\n",
       "      <td>1.349565</td>\n",
       "      <td>1.066702</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>475 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID APPRDX2  CAUDATE_R  CAUDATE_L  PUTAMEN_R  PUTAMEN_L\n",
       "0    3000  3000      HC       2.99       3.43   2.328620   1.973532\n",
       "1    3001  3001      PD       1.56       1.12  -0.246956  -0.796382\n",
       "2    3002  3002      PD       2.13       2.28   0.255480   0.339528\n",
       "3    3004  3004      HC       5.09       5.30   2.949289   2.400680\n",
       "4    3006  3006      PD       2.28       2.12   0.175997  -2.400680\n",
       "..    ...   ...     ...        ...        ...        ...        ...\n",
       "470  4124  4124      PD       1.95       1.59  -0.273727  -0.053985\n",
       "471  4125  4125      PD       2.03       2.47   0.391926   1.005487\n",
       "472  4126  4126      PD       1.54       1.88  -0.219232  -1.158726\n",
       "473  4136  4136      PD       0.68       0.57  -1.406477  -1.922162\n",
       "474  4139  4139      HC       2.44       2.82   1.349565   1.066702\n",
       "\n",
       "[475 rows x 7 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno = info_pheno[[\"FID\",\"IID\",\"APPRDX2\",\"CAUDATE_R\",\"CAUDATE_L\",\"PUTAMEN_R\",\"PUTAMEN_L\"]]\n",
    "pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno.to_csv('GWAS2/Phenotype.txt',sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "age = list()\n",
    "\n",
    "for i in range(len(info_pheno)):\n",
    "    mm,yy = info_pheno.loc[i,\"SITE_APRV\"].split(\"/\")\n",
    "    a = int(yy) - int(info_pheno.loc[i,\"BIRTHDT\"])\n",
    "    age.append(a)\n",
    "\n",
    "info_pheno[\"AGE\"] = age   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>70</td>\n",
       "      <td>0.021562</td>\n",
       "      <td>-0.005305</td>\n",
       "      <td>0.012287</td>\n",
       "      <td>0.065969</td>\n",
       "      <td>0.006345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>65</td>\n",
       "      <td>0.013209</td>\n",
       "      <td>0.016830</td>\n",
       "      <td>0.014295</td>\n",
       "      <td>-0.033689</td>\n",
       "      <td>-0.039806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>68</td>\n",
       "      <td>-0.082064</td>\n",
       "      <td>-0.030599</td>\n",
       "      <td>-0.021643</td>\n",
       "      <td>-0.001498</td>\n",
       "      <td>0.025769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>60</td>\n",
       "      <td>0.023291</td>\n",
       "      <td>0.022143</td>\n",
       "      <td>-0.013579</td>\n",
       "      <td>-0.013314</td>\n",
       "      <td>-0.029992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>58</td>\n",
       "      <td>0.008937</td>\n",
       "      <td>0.057132</td>\n",
       "      <td>0.088903</td>\n",
       "      <td>0.031578</td>\n",
       "      <td>0.002286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>71</td>\n",
       "      <td>0.007515</td>\n",
       "      <td>0.030850</td>\n",
       "      <td>0.032300</td>\n",
       "      <td>0.006790</td>\n",
       "      <td>0.075679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>64</td>\n",
       "      <td>-0.015873</td>\n",
       "      <td>0.026348</td>\n",
       "      <td>-0.005282</td>\n",
       "      <td>-0.022791</td>\n",
       "      <td>-0.076218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>56</td>\n",
       "      <td>-0.006236</td>\n",
       "      <td>-0.004422</td>\n",
       "      <td>-0.054059</td>\n",
       "      <td>0.034937</td>\n",
       "      <td>0.011776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>56</td>\n",
       "      <td>0.036144</td>\n",
       "      <td>-0.015858</td>\n",
       "      <td>-0.064136</td>\n",
       "      <td>-0.033492</td>\n",
       "      <td>0.042224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>81</td>\n",
       "      <td>0.038298</td>\n",
       "      <td>-0.099336</td>\n",
       "      <td>-0.061020</td>\n",
       "      <td>-0.021126</td>\n",
       "      <td>0.006314</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>475 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID  AGE       PC1       PC2       PC3       PC4       PC5\n",
       "0    3000  3000   70  0.021562 -0.005305  0.012287  0.065969  0.006345\n",
       "1    3001  3001   65  0.013209  0.016830  0.014295 -0.033689 -0.039806\n",
       "2    3002  3002   68 -0.082064 -0.030599 -0.021643 -0.001498  0.025769\n",
       "3    3004  3004   60  0.023291  0.022143 -0.013579 -0.013314 -0.029992\n",
       "4    3006  3006   58  0.008937  0.057132  0.088903  0.031578  0.002286\n",
       "..    ...   ...  ...       ...       ...       ...       ...       ...\n",
       "470  4124  4124   71  0.007515  0.030850  0.032300  0.006790  0.075679\n",
       "471  4125  4125   64 -0.015873  0.026348 -0.005282 -0.022791 -0.076218\n",
       "472  4126  4126   56 -0.006236 -0.004422 -0.054059  0.034937  0.011776\n",
       "473  4136  4136   56  0.036144 -0.015858 -0.064136 -0.033492  0.042224\n",
       "474  4139  4139   81  0.038298 -0.099336 -0.061020 -0.021126  0.006314\n",
       "\n",
       "[475 rows x 8 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = info_pheno[[\"FID\",\"IID\",\"AGE\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\"]]\n",
    "cov.to_csv('GWAS2/Covariate.txt',sep=\"\\t\", index=False)\n",
    "cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CAUDATE_R', 'CAUDATE_L', 'PUTAMEN_R', 'PUTAMEN_L']"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col = list(pheno.columns.values)[3:7] \n",
    "col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/CAUDATE_R.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/CAUDATE_R\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name CAUDATE_R\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "475 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/CAUDATE_R.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/CAUDATE_R.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/CAUDATE_L.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/CAUDATE_L\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name CAUDATE_L\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "475 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/CAUDATE_L.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.01682.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/CAUDATE_L.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/PUTAMEN_R.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/PUTAMEN_R\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name PUTAMEN_R\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "475 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/PUTAMEN_R.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/PUTAMEN_R.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/PUTAMEN_L.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/PUTAMEN_L\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name PUTAMEN_L\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "475 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/PUTAMEN_L.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/PUTAMEN_L.assoc.linear.adjusted .\n"
     ]
    }
   ],
   "source": [
    "for i in col:\n",
    "    !plink --bfile DATA/QC/White_2 --pheno GWAS2/Phenotype.txt --covar GWAS2/Covariate.txt --pheno-name {i} --covar-name AGE,PC1-PC5 --linear sex --adjust --out GWAS2/{i}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in col:\n",
    "    gwas = pd.read_table(\"GWAS2/\"+i+\".assoc.linear\",sep=\"\\s+\")\n",
    "    gwas = gwas[gwas.TEST==\"ADD\"]\n",
    "    gwas.to_csv('GWAS2/GWAS_DATScan_'+i+'.csv',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Datscan GWAS without SWEDD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>REFERRAL</th>\n",
       "      <th>DECLINED</th>\n",
       "      <th>RSNDEC</th>\n",
       "      <th>EXCLUDED</th>\n",
       "      <th>RSNEXC</th>\n",
       "      <th>ORIG_ENTRY</th>\n",
       "      <th>LAST_UPDATE</th>\n",
       "      <th>QUERY</th>\n",
       "      <th>SITE_APRV</th>\n",
       "      <th>APPRDX2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>0.026492</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>0.035338</td>\n",
       "      <td>0.064825</td>\n",
       "      <td>0.000766</td>\n",
       "      <td>-0.041875</td>\n",
       "      <td>0.033775</td>\n",
       "      <td>0.013332</td>\n",
       "      <td>...</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>2011-01-19 12:44:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>0.017766</td>\n",
       "      <td>0.017305</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>-0.045234</td>\n",
       "      <td>-0.037821</td>\n",
       "      <td>-0.066630</td>\n",
       "      <td>0.010062</td>\n",
       "      <td>-0.054224</td>\n",
       "      <td>...</td>\n",
       "      <td>60.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2011</td>\n",
       "      <td>2011-02-10 10:43:30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>-0.084263</td>\n",
       "      <td>-0.034856</td>\n",
       "      <td>-0.023204</td>\n",
       "      <td>0.006116</td>\n",
       "      <td>0.027577</td>\n",
       "      <td>0.008214</td>\n",
       "      <td>-0.006264</td>\n",
       "      <td>0.008593</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-04 11:04:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>0.024563</td>\n",
       "      <td>0.033477</td>\n",
       "      <td>-0.003634</td>\n",
       "      <td>-0.005703</td>\n",
       "      <td>-0.042231</td>\n",
       "      <td>-0.059694</td>\n",
       "      <td>0.067412</td>\n",
       "      <td>0.017923</td>\n",
       "      <td>...</td>\n",
       "      <td>99.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-30 11:43:13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>0.008511</td>\n",
       "      <td>0.056539</td>\n",
       "      <td>0.101538</td>\n",
       "      <td>0.001351</td>\n",
       "      <td>0.006814</td>\n",
       "      <td>-0.047180</td>\n",
       "      <td>-0.012988</td>\n",
       "      <td>-0.033577</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-31 07:21:03.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>0.008077</td>\n",
       "      <td>0.029449</td>\n",
       "      <td>0.040843</td>\n",
       "      <td>-0.013005</td>\n",
       "      <td>0.074609</td>\n",
       "      <td>0.068977</td>\n",
       "      <td>0.010669</td>\n",
       "      <td>0.081488</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-02-22 12:57:46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>-0.019947</td>\n",
       "      <td>0.029816</td>\n",
       "      <td>-0.017732</td>\n",
       "      <td>-0.017475</td>\n",
       "      <td>-0.082603</td>\n",
       "      <td>0.015329</td>\n",
       "      <td>0.025788</td>\n",
       "      <td>0.053534</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-12-10 12:36:00.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>05/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>-0.009423</td>\n",
       "      <td>0.001673</td>\n",
       "      <td>-0.037664</td>\n",
       "      <td>0.060811</td>\n",
       "      <td>-0.006292</td>\n",
       "      <td>0.034639</td>\n",
       "      <td>0.107867</td>\n",
       "      <td>0.040532</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-27 08:35:42.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.032763</td>\n",
       "      <td>-0.016046</td>\n",
       "      <td>-0.064461</td>\n",
       "      <td>-0.015332</td>\n",
       "      <td>0.030023</td>\n",
       "      <td>0.123020</td>\n",
       "      <td>0.032174</td>\n",
       "      <td>0.084627</td>\n",
       "      <td>...</td>\n",
       "      <td>99.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-06 07:09:02.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>PD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>0.048842</td>\n",
       "      <td>-0.100729</td>\n",
       "      <td>-0.075878</td>\n",
       "      <td>-0.009114</td>\n",
       "      <td>0.002229</td>\n",
       "      <td>0.052469</td>\n",
       "      <td>-0.000137</td>\n",
       "      <td>0.078665</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-05-16 13:26:16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>HC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>436 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID       PC1       PC2       PC3       PC4       PC5       PC6  \\\n",
       "0    3000  3000  0.026492  0.000678  0.035338  0.064825  0.000766 -0.041875   \n",
       "1    3001  3001  0.017766  0.017305  0.000688 -0.045234 -0.037821 -0.066630   \n",
       "2    3002  3002 -0.084263 -0.034856 -0.023204  0.006116  0.027577  0.008214   \n",
       "3    3004  3004  0.024563  0.033477 -0.003634 -0.005703 -0.042231 -0.059694   \n",
       "4    3006  3006  0.008511  0.056539  0.101538  0.001351  0.006814 -0.047180   \n",
       "..    ...   ...       ...       ...       ...       ...       ...       ...   \n",
       "431  4124  4124  0.008077  0.029449  0.040843 -0.013005  0.074609  0.068977   \n",
       "432  4125  4125 -0.019947  0.029816 -0.017732 -0.017475 -0.082603  0.015329   \n",
       "433  4126  4126 -0.009423  0.001673 -0.037664  0.060811 -0.006292  0.034639   \n",
       "434  4136  4136  0.032763 -0.016046 -0.064461 -0.015332  0.030023  0.123020   \n",
       "435  4139  4139  0.048842 -0.100729 -0.075878 -0.009114  0.002229  0.052469   \n",
       "\n",
       "          PC7       PC8  ...  REFERRAL  DECLINED  RSNDEC EXCLUDED  RSNEXC  \\\n",
       "0    0.033775  0.013332  ...      31.0       NaN     NaN      NaN     NaN   \n",
       "1    0.010062 -0.054224  ...      60.0       NaN     NaN      NaN     NaN   \n",
       "2   -0.006264  0.008593  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "3    0.067412  0.017923  ...      99.0       NaN     NaN      NaN     NaN   \n",
       "4   -0.012988 -0.033577  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "..        ...       ...  ...       ...       ...     ...      ...     ...   \n",
       "431  0.010669  0.081488  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "432  0.025788  0.053534  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "433  0.107867  0.040532  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "434  0.032174  0.084627  ...      99.0       NaN     NaN      NaN     NaN   \n",
       "435 -0.000137  0.078665  ...       1.0       NaN     NaN      NaN     NaN   \n",
       "\n",
       "    ORIG_ENTRY            LAST_UPDATE  QUERY SITE_APRV  APPRDX2  \n",
       "0      01/2011  2011-01-19 12:44:58.0    NaN   01/2011       HC  \n",
       "1      02/2011  2011-02-10 10:43:30.0    NaN   03/2011       PD  \n",
       "2      03/2011  2011-03-04 11:04:58.0    NaN   03/2011       PD  \n",
       "3      03/2011  2011-03-30 11:43:13.0    NaN   03/2011       HC  \n",
       "4      03/2011  2011-03-31 07:21:03.0    NaN   03/2011       PD  \n",
       "..         ...                    ...    ...       ...      ...  \n",
       "431    02/2013  2013-02-22 12:57:46.0    NaN   02/2013       PD  \n",
       "432    02/2013  2013-12-10 12:36:00.0    NaN   05/2013       PD  \n",
       "433    03/2013  2013-03-27 08:35:42.0    NaN   03/2013       PD  \n",
       "434    03/2013  2013-03-06 07:09:02.0    NaN   06/2013       PD  \n",
       "435    03/2013  2013-05-16 13:26:16.0    NaN   06/2013       HC  \n",
       "\n",
       "[436 rows x 42 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pca = pd.read_table('Data/QC/White_no_swedd.eigenvec', sep=' ', header=None)\n",
    "cnames = ['FID','IID']\n",
    "for i in range(1,11):\n",
    "    cnames.append(\"PC\"+str(i))\n",
    "pca.columns = cnames\n",
    "\n",
    "info_pca = pca.merge(ppmi_info, left_on=['FID'], right_on=['PATNO'])\n",
    "\n",
    "apprdx = list()\n",
    "\n",
    "for i in info_pca['APPRDX'].to_list():\n",
    "    if(i==1):\n",
    "        apprdx.append(\"PD\")\n",
    "    if(i==2):\n",
    "        apprdx.append(\"HC\")\n",
    "    elif(i==3):\n",
    "        apprdx.append(\"SWEDD\")\n",
    "\n",
    "info_pca['APPRDX2'] = apprdx\n",
    "info_pca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>ORIG_ENTRY</th>\n",
       "      <th>LAST_UPDATE</th>\n",
       "      <th>QUERY</th>\n",
       "      <th>SITE_APRV</th>\n",
       "      <th>APPRDX2</th>\n",
       "      <th>PATNO_y</th>\n",
       "      <th>CAUDATE_R</th>\n",
       "      <th>CAUDATE_L</th>\n",
       "      <th>PUTAMEN_R</th>\n",
       "      <th>PUTAMEN_L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>0.026492</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>0.035338</td>\n",
       "      <td>0.064825</td>\n",
       "      <td>0.000766</td>\n",
       "      <td>-0.041875</td>\n",
       "      <td>0.033775</td>\n",
       "      <td>0.013332</td>\n",
       "      <td>...</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>2011-01-19 12:44:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>01/2011</td>\n",
       "      <td>HC</td>\n",
       "      <td>3000</td>\n",
       "      <td>2.99</td>\n",
       "      <td>3.43</td>\n",
       "      <td>2.328620</td>\n",
       "      <td>1.973532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>0.017766</td>\n",
       "      <td>0.017305</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>-0.045234</td>\n",
       "      <td>-0.037821</td>\n",
       "      <td>-0.066630</td>\n",
       "      <td>0.010062</td>\n",
       "      <td>-0.054224</td>\n",
       "      <td>...</td>\n",
       "      <td>02/2011</td>\n",
       "      <td>2011-02-10 10:43:30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "      <td>3001</td>\n",
       "      <td>1.56</td>\n",
       "      <td>1.12</td>\n",
       "      <td>-0.246956</td>\n",
       "      <td>-0.796382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>-0.084263</td>\n",
       "      <td>-0.034856</td>\n",
       "      <td>-0.023204</td>\n",
       "      <td>0.006116</td>\n",
       "      <td>0.027577</td>\n",
       "      <td>0.008214</td>\n",
       "      <td>-0.006264</td>\n",
       "      <td>0.008593</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-04 11:04:58.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "      <td>3002</td>\n",
       "      <td>2.13</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.255480</td>\n",
       "      <td>0.339528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>0.024563</td>\n",
       "      <td>0.033477</td>\n",
       "      <td>-0.003634</td>\n",
       "      <td>-0.005703</td>\n",
       "      <td>-0.042231</td>\n",
       "      <td>-0.059694</td>\n",
       "      <td>0.067412</td>\n",
       "      <td>0.017923</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-30 11:43:13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>HC</td>\n",
       "      <td>3004</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.30</td>\n",
       "      <td>2.949289</td>\n",
       "      <td>2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>0.008511</td>\n",
       "      <td>0.056539</td>\n",
       "      <td>0.101538</td>\n",
       "      <td>0.001351</td>\n",
       "      <td>0.006814</td>\n",
       "      <td>-0.047180</td>\n",
       "      <td>-0.012988</td>\n",
       "      <td>-0.033577</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>2011-03-31 07:21:03.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2011</td>\n",
       "      <td>PD</td>\n",
       "      <td>3006</td>\n",
       "      <td>2.28</td>\n",
       "      <td>2.12</td>\n",
       "      <td>0.175997</td>\n",
       "      <td>-2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>0.008077</td>\n",
       "      <td>0.029449</td>\n",
       "      <td>0.040843</td>\n",
       "      <td>-0.013005</td>\n",
       "      <td>0.074609</td>\n",
       "      <td>0.068977</td>\n",
       "      <td>0.010669</td>\n",
       "      <td>0.081488</td>\n",
       "      <td>...</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-02-22 12:57:46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4124</td>\n",
       "      <td>1.95</td>\n",
       "      <td>1.59</td>\n",
       "      <td>-0.273727</td>\n",
       "      <td>-0.053985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>-0.019947</td>\n",
       "      <td>0.029816</td>\n",
       "      <td>-0.017732</td>\n",
       "      <td>-0.017475</td>\n",
       "      <td>-0.082603</td>\n",
       "      <td>0.015329</td>\n",
       "      <td>0.025788</td>\n",
       "      <td>0.053534</td>\n",
       "      <td>...</td>\n",
       "      <td>02/2013</td>\n",
       "      <td>2013-12-10 12:36:00.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>05/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4125</td>\n",
       "      <td>2.03</td>\n",
       "      <td>2.47</td>\n",
       "      <td>0.391926</td>\n",
       "      <td>1.005487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>-0.009423</td>\n",
       "      <td>0.001673</td>\n",
       "      <td>-0.037664</td>\n",
       "      <td>0.060811</td>\n",
       "      <td>-0.006292</td>\n",
       "      <td>0.034639</td>\n",
       "      <td>0.107867</td>\n",
       "      <td>0.040532</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-27 08:35:42.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4126</td>\n",
       "      <td>1.54</td>\n",
       "      <td>1.88</td>\n",
       "      <td>-0.219232</td>\n",
       "      <td>-1.158726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.032763</td>\n",
       "      <td>-0.016046</td>\n",
       "      <td>-0.064461</td>\n",
       "      <td>-0.015332</td>\n",
       "      <td>0.030023</td>\n",
       "      <td>0.123020</td>\n",
       "      <td>0.032174</td>\n",
       "      <td>0.084627</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-03-06 07:09:02.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>PD</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.57</td>\n",
       "      <td>-1.406477</td>\n",
       "      <td>-1.922162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>0.048842</td>\n",
       "      <td>-0.100729</td>\n",
       "      <td>-0.075878</td>\n",
       "      <td>-0.009114</td>\n",
       "      <td>0.002229</td>\n",
       "      <td>0.052469</td>\n",
       "      <td>-0.000137</td>\n",
       "      <td>0.078665</td>\n",
       "      <td>...</td>\n",
       "      <td>03/2013</td>\n",
       "      <td>2013-05-16 13:26:16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06/2013</td>\n",
       "      <td>HC</td>\n",
       "      <td>4139</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.82</td>\n",
       "      <td>1.349565</td>\n",
       "      <td>1.066702</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID       PC1       PC2       PC3       PC4       PC5       PC6  \\\n",
       "0    3000  3000  0.026492  0.000678  0.035338  0.064825  0.000766 -0.041875   \n",
       "1    3001  3001  0.017766  0.017305  0.000688 -0.045234 -0.037821 -0.066630   \n",
       "2    3002  3002 -0.084263 -0.034856 -0.023204  0.006116  0.027577  0.008214   \n",
       "3    3004  3004  0.024563  0.033477 -0.003634 -0.005703 -0.042231 -0.059694   \n",
       "4    3006  3006  0.008511  0.056539  0.101538  0.001351  0.006814 -0.047180   \n",
       "..    ...   ...       ...       ...       ...       ...       ...       ...   \n",
       "428  4124  4124  0.008077  0.029449  0.040843 -0.013005  0.074609  0.068977   \n",
       "429  4125  4125 -0.019947  0.029816 -0.017732 -0.017475 -0.082603  0.015329   \n",
       "430  4126  4126 -0.009423  0.001673 -0.037664  0.060811 -0.006292  0.034639   \n",
       "431  4136  4136  0.032763 -0.016046 -0.064461 -0.015332  0.030023  0.123020   \n",
       "432  4139  4139  0.048842 -0.100729 -0.075878 -0.009114  0.002229  0.052469   \n",
       "\n",
       "          PC7       PC8  ...  ORIG_ENTRY            LAST_UPDATE  QUERY  \\\n",
       "0    0.033775  0.013332  ...     01/2011  2011-01-19 12:44:58.0    NaN   \n",
       "1    0.010062 -0.054224  ...     02/2011  2011-02-10 10:43:30.0    NaN   \n",
       "2   -0.006264  0.008593  ...     03/2011  2011-03-04 11:04:58.0    NaN   \n",
       "3    0.067412  0.017923  ...     03/2011  2011-03-30 11:43:13.0    NaN   \n",
       "4   -0.012988 -0.033577  ...     03/2011  2011-03-31 07:21:03.0    NaN   \n",
       "..        ...       ...  ...         ...                    ...    ...   \n",
       "428  0.010669  0.081488  ...     02/2013  2013-02-22 12:57:46.0    NaN   \n",
       "429  0.025788  0.053534  ...     02/2013  2013-12-10 12:36:00.0    NaN   \n",
       "430  0.107867  0.040532  ...     03/2013  2013-03-27 08:35:42.0    NaN   \n",
       "431  0.032174  0.084627  ...     03/2013  2013-03-06 07:09:02.0    NaN   \n",
       "432 -0.000137  0.078665  ...     03/2013  2013-05-16 13:26:16.0    NaN   \n",
       "\n",
       "    SITE_APRV  APPRDX2 PATNO_y CAUDATE_R  CAUDATE_L PUTAMEN_R  PUTAMEN_L  \n",
       "0     01/2011       HC    3000      2.99       3.43  2.328620   1.973532  \n",
       "1     03/2011       PD    3001      1.56       1.12 -0.246956  -0.796382  \n",
       "2     03/2011       PD    3002      2.13       2.28  0.255480   0.339528  \n",
       "3     03/2011       HC    3004      5.09       5.30  2.949289   2.400680  \n",
       "4     03/2011       PD    3006      2.28       2.12  0.175997  -2.400680  \n",
       "..        ...      ...     ...       ...        ...       ...        ...  \n",
       "428   02/2013       PD    4124      1.95       1.59 -0.273727  -0.053985  \n",
       "429   05/2013       PD    4125      2.03       2.47  0.391926   1.005487  \n",
       "430   03/2013       PD    4126      1.54       1.88 -0.219232  -1.158726  \n",
       "431   06/2013       PD    4136      0.68       0.57 -1.406477  -1.922162  \n",
       "432   06/2013       HC    4139      2.44       2.82  1.349565   1.066702  \n",
       "\n",
       "[433 rows x 47 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_pheno = info_pca.merge(images, left_on=['FID'], right_on=['PATNO'])\n",
    "info_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to Data/QC/White_2_no_swedd.log.\n",
      "Options in effect:\n",
      "  --bfile Data/QC/White_no_swedd\n",
      "  --keep Data/QC/IDs_White_no_swedd.txt\n",
      "  --make-bed\n",
      "  --out Data/QC/White_2_no_swedd\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "436 people (289 males, 147 females) loaded from .fam.\n",
      "--keep: 433 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to Data/QC/White_2_no_swedd.bed + Data/QC/White_2_no_swedd.bim +\n",
      "Data/QC/White_2_no_swedd.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "info_pheno.iloc[:,0:2].to_csv(\"Data/QC/IDs_White_no_swedd.txt\", header=False, index=False, sep=\" \")\n",
    "!plink --bfile Data/QC/White_no_swedd --keep Data/QC/IDs_White_no_swedd.txt --make-bed --out Data/QC/White_2_no_swedd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>APPRDX2</th>\n",
       "      <th>CAUDATE_R</th>\n",
       "      <th>CAUDATE_L</th>\n",
       "      <th>PUTAMEN_R</th>\n",
       "      <th>PUTAMEN_L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.99</td>\n",
       "      <td>3.43</td>\n",
       "      <td>2.328620</td>\n",
       "      <td>1.973532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.56</td>\n",
       "      <td>1.12</td>\n",
       "      <td>-0.246956</td>\n",
       "      <td>-0.796382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>PD</td>\n",
       "      <td>2.13</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.255480</td>\n",
       "      <td>0.339528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>HC</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.30</td>\n",
       "      <td>2.949289</td>\n",
       "      <td>2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>PD</td>\n",
       "      <td>2.28</td>\n",
       "      <td>2.12</td>\n",
       "      <td>0.175997</td>\n",
       "      <td>-2.400680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.95</td>\n",
       "      <td>1.59</td>\n",
       "      <td>-0.273727</td>\n",
       "      <td>-0.053985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>PD</td>\n",
       "      <td>2.03</td>\n",
       "      <td>2.47</td>\n",
       "      <td>0.391926</td>\n",
       "      <td>1.005487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>PD</td>\n",
       "      <td>1.54</td>\n",
       "      <td>1.88</td>\n",
       "      <td>-0.219232</td>\n",
       "      <td>-1.158726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>PD</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.57</td>\n",
       "      <td>-1.406477</td>\n",
       "      <td>-1.922162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.82</td>\n",
       "      <td>1.349565</td>\n",
       "      <td>1.066702</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID APPRDX2  CAUDATE_R  CAUDATE_L  PUTAMEN_R  PUTAMEN_L\n",
       "0    3000  3000      HC       2.99       3.43   2.328620   1.973532\n",
       "1    3001  3001      PD       1.56       1.12  -0.246956  -0.796382\n",
       "2    3002  3002      PD       2.13       2.28   0.255480   0.339528\n",
       "3    3004  3004      HC       5.09       5.30   2.949289   2.400680\n",
       "4    3006  3006      PD       2.28       2.12   0.175997  -2.400680\n",
       "..    ...   ...     ...        ...        ...        ...        ...\n",
       "428  4124  4124      PD       1.95       1.59  -0.273727  -0.053985\n",
       "429  4125  4125      PD       2.03       2.47   0.391926   1.005487\n",
       "430  4126  4126      PD       1.54       1.88  -0.219232  -1.158726\n",
       "431  4136  4136      PD       0.68       0.57  -1.406477  -1.922162\n",
       "432  4139  4139      HC       2.44       2.82   1.349565   1.066702\n",
       "\n",
       "[433 rows x 7 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno = info_pheno[[\"FID\",\"IID\",\"APPRDX2\",\"CAUDATE_R\",\"CAUDATE_L\",\"PUTAMEN_R\",\"PUTAMEN_L\"]]\n",
    "pheno.to_csv('GWAS2/Phenotype.txt',sep=\"\\t\", index=False)\n",
    "pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>70</td>\n",
       "      <td>0.026492</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>0.035338</td>\n",
       "      <td>0.064825</td>\n",
       "      <td>0.000766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>65</td>\n",
       "      <td>0.017766</td>\n",
       "      <td>0.017305</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>-0.045234</td>\n",
       "      <td>-0.037821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>68</td>\n",
       "      <td>-0.084263</td>\n",
       "      <td>-0.034856</td>\n",
       "      <td>-0.023204</td>\n",
       "      <td>0.006116</td>\n",
       "      <td>0.027577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>60</td>\n",
       "      <td>0.024563</td>\n",
       "      <td>0.033477</td>\n",
       "      <td>-0.003634</td>\n",
       "      <td>-0.005703</td>\n",
       "      <td>-0.042231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>58</td>\n",
       "      <td>0.008511</td>\n",
       "      <td>0.056539</td>\n",
       "      <td>0.101538</td>\n",
       "      <td>0.001351</td>\n",
       "      <td>0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>71</td>\n",
       "      <td>0.008077</td>\n",
       "      <td>0.029449</td>\n",
       "      <td>0.040843</td>\n",
       "      <td>-0.013005</td>\n",
       "      <td>0.074609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>64</td>\n",
       "      <td>-0.019947</td>\n",
       "      <td>0.029816</td>\n",
       "      <td>-0.017732</td>\n",
       "      <td>-0.017475</td>\n",
       "      <td>-0.082603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>56</td>\n",
       "      <td>-0.009423</td>\n",
       "      <td>0.001673</td>\n",
       "      <td>-0.037664</td>\n",
       "      <td>0.060811</td>\n",
       "      <td>-0.006292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>56</td>\n",
       "      <td>0.032763</td>\n",
       "      <td>-0.016046</td>\n",
       "      <td>-0.064461</td>\n",
       "      <td>-0.015332</td>\n",
       "      <td>0.030023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>81</td>\n",
       "      <td>0.048842</td>\n",
       "      <td>-0.100729</td>\n",
       "      <td>-0.075878</td>\n",
       "      <td>-0.009114</td>\n",
       "      <td>0.002229</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID  AGE       PC1       PC2       PC3       PC4       PC5\n",
       "0    3000  3000   70  0.026492  0.000678  0.035338  0.064825  0.000766\n",
       "1    3001  3001   65  0.017766  0.017305  0.000688 -0.045234 -0.037821\n",
       "2    3002  3002   68 -0.084263 -0.034856 -0.023204  0.006116  0.027577\n",
       "3    3004  3004   60  0.024563  0.033477 -0.003634 -0.005703 -0.042231\n",
       "4    3006  3006   58  0.008511  0.056539  0.101538  0.001351  0.006814\n",
       "..    ...   ...  ...       ...       ...       ...       ...       ...\n",
       "428  4124  4124   71  0.008077  0.029449  0.040843 -0.013005  0.074609\n",
       "429  4125  4125   64 -0.019947  0.029816 -0.017732 -0.017475 -0.082603\n",
       "430  4126  4126   56 -0.009423  0.001673 -0.037664  0.060811 -0.006292\n",
       "431  4136  4136   56  0.032763 -0.016046 -0.064461 -0.015332  0.030023\n",
       "432  4139  4139   81  0.048842 -0.100729 -0.075878 -0.009114  0.002229\n",
       "\n",
       "[433 rows x 8 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age = list()\n",
    "\n",
    "for i in range(len(info_pheno)):\n",
    "    mm,yy = info_pheno.loc[i,\"SITE_APRV\"].split(\"/\")\n",
    "    a = int(yy) - int(info_pheno.loc[i,\"BIRTHDT\"])\n",
    "    age.append(a)\n",
    "\n",
    "info_pheno[\"AGE\"] = age   \n",
    "\n",
    "cov = info_pheno[[\"FID\",\"IID\",\"AGE\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\"]]\n",
    "cov.to_csv('GWAS2/Covariate.txt',sep=\"\\t\", index=False)\n",
    "cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CAUDATE_R', 'CAUDATE_L', 'PUTAMEN_R', 'PUTAMEN_L']"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col = list(pheno.columns.values)[3:7] \n",
    "col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/CAUDATE_R.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/CAUDATE_R\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name CAUDATE_R\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "433 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/CAUDATE_R.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/CAUDATE_R.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/CAUDATE_L.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/CAUDATE_L\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name CAUDATE_L\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "433 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/CAUDATE_L.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/CAUDATE_L.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/PUTAMEN_R.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/PUTAMEN_R\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name PUTAMEN_R\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "433 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/PUTAMEN_R.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/PUTAMEN_R.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/PUTAMEN_L.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/Covariate.txt\n",
      "  --covar-name AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/PUTAMEN_L\n",
      "  --pheno GWAS2/Phenotype.txt\n",
      "  --pheno-name PUTAMEN_L\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "433 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 6 covariates loaded.\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to GWAS2/PUTAMEN_L.assoc.linear ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/PUTAMEN_L.assoc.linear.adjusted .\n"
     ]
    }
   ],
   "source": [
    "for i in col:\n",
    "    !plink --bfile DATA/QC/White_2_no_swedd --pheno GWAS2/Phenotype.txt --covar GWAS2/Covariate.txt --pheno-name {i} --covar-name AGE,PC1-PC5 --linear sex --adjust --out GWAS2/{i}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in col:\n",
    "    gwas = pd.read_table(\"GWAS2/\"+i+\".assoc.linear\",sep=\"\\s+\")\n",
    "    gwas = gwas[gwas.TEST==\"ADD\"]\n",
    "    gwas.to_csv('GWAS2/GWAS_DATScan_'+i+'.csv',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MRI GWAS WITH SWEDD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/guglielmo/opt/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3063: DtypeWarning: Columns (31,494,496,506,508,510,512,542,543,550,551,553,561,563,564) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATNO</th>\n",
       "      <th>index</th>\n",
       "      <th>APPRDX_enrol</th>\n",
       "      <th>APPRDX_current</th>\n",
       "      <th>EVENT_ID</th>\n",
       "      <th>Age</th>\n",
       "      <th>Years_bl</th>\n",
       "      <th>GENDER</th>\n",
       "      <th>FAMHXPD_N1stDegree</th>\n",
       "      <th>FAMHXPD_N2ndDegree</th>\n",
       "      <th>...</th>\n",
       "      <th>TotalGrayVol</th>\n",
       "      <th>SupraTentorialVol</th>\n",
       "      <th>SupraTentorialVolNotVent</th>\n",
       "      <th>SupraTentorialVolNotVentVox</th>\n",
       "      <th>TD_score</th>\n",
       "      <th>PIGD_score</th>\n",
       "      <th>TD_PIGD_ratio</th>\n",
       "      <th>TD_PIGD_class</th>\n",
       "      <th>INFODT_</th>\n",
       "      <th>INFODT_bl</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>651</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>69.169</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>628301.286475</td>\n",
       "      <td>9.996803e+05</td>\n",
       "      <td>9.777723e+05</td>\n",
       "      <td>973296.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-02-01</td>\n",
       "      <td>2011-02-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>687</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>65.161</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>668025.901257</td>\n",
       "      <td>1.138365e+06</td>\n",
       "      <td>1.094811e+06</td>\n",
       "      <td>1092459.0</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>TD</td>\n",
       "      <td>2011-03-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>759</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>67.581</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>585538.400359</td>\n",
       "      <td>9.026203e+05</td>\n",
       "      <td>8.838913e+05</td>\n",
       "      <td>881444.0</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2011-03-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3003</td>\n",
       "      <td>831</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>56.750</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>636432.509323</td>\n",
       "      <td>1.040302e+06</td>\n",
       "      <td>1.020943e+06</td>\n",
       "      <td>1019030.0</td>\n",
       "      <td>0.363636</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>2011-04-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3004</td>\n",
       "      <td>867</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>59.414</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>640446.950904</td>\n",
       "      <td>1.074093e+06</td>\n",
       "      <td>1.056334e+06</td>\n",
       "      <td>1053446.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>2011-04-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1906</th>\n",
       "      <td>90456</td>\n",
       "      <td>69726</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>82.505</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>635113.067329</td>\n",
       "      <td>1.123903e+06</td>\n",
       "      <td>1.030924e+06</td>\n",
       "      <td>1027319.0</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.151515</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2013-09-01</td>\n",
       "      <td>2013-09-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1907</th>\n",
       "      <td>91097</td>\n",
       "      <td>69762</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>67.335</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>507261.673333</td>\n",
       "      <td>8.354743e+05</td>\n",
       "      <td>8.187873e+05</td>\n",
       "      <td>816977.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2014-07-01</td>\n",
       "      <td>2014-07-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1908</th>\n",
       "      <td>91837</td>\n",
       "      <td>69798</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>73.914</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2014-03-01</td>\n",
       "      <td>2014-03-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1909</th>\n",
       "      <td>92490</td>\n",
       "      <td>69834</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>72.504</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>TD</td>\n",
       "      <td>2015-01-01</td>\n",
       "      <td>2015-01-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1910</th>\n",
       "      <td>92834</td>\n",
       "      <td>69870</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>BL</td>\n",
       "      <td>66.168</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>619843.427029</td>\n",
       "      <td>1.001420e+06</td>\n",
       "      <td>9.639952e+05</td>\n",
       "      <td>962387.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2015-02-01</td>\n",
       "      <td>2015-02-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1911 rows × 903 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      PATNO  index  APPRDX_enrol  APPRDX_current EVENT_ID     Age  Years_bl  \\\n",
       "0      3000    651           2.0             2.0       BL  69.169       0.0   \n",
       "1      3001    687           1.0             1.0       BL  65.161       NaN   \n",
       "2      3002    759           1.0             1.0       BL  67.581       NaN   \n",
       "3      3003    831           1.0             1.0       BL  56.750       0.0   \n",
       "4      3004    867           2.0             2.0       BL  59.414       0.0   \n",
       "...     ...    ...           ...             ...      ...     ...       ...   \n",
       "1906  90456  69726           4.0             4.0       BL  82.505       0.0   \n",
       "1907  91097  69762           4.0             4.0       BL  67.335       0.0   \n",
       "1908  91837  69798           4.0             4.0       BL  73.914       0.0   \n",
       "1909  92490  69834           4.0             4.0       BL  72.504       0.0   \n",
       "1910  92834  69870           4.0             4.0       BL  66.168       0.0   \n",
       "\n",
       "      GENDER  FAMHXPD_N1stDegree  FAMHXPD_N2ndDegree  ...   TotalGrayVol  \\\n",
       "0        1.0                 0.0                 0.0  ...  628301.286475   \n",
       "1        2.0                 0.0                 0.0  ...  668025.901257   \n",
       "2        1.0                 0.0                 0.0  ...  585538.400359   \n",
       "3        1.0                 0.0                 0.0  ...  636432.509323   \n",
       "4        2.0                 0.0                 0.0  ...  640446.950904   \n",
       "...      ...                 ...                 ...  ...            ...   \n",
       "1906     2.0                 0.0                 0.0  ...  635113.067329   \n",
       "1907     2.0                 0.0                 0.0  ...  507261.673333   \n",
       "1908     2.0                 0.0                 0.0  ...            NaN   \n",
       "1909     2.0                 0.0                 0.0  ...            NaN   \n",
       "1910     2.0                 0.0                 0.0  ...  619843.427029   \n",
       "\n",
       "     SupraTentorialVol  SupraTentorialVolNotVent  SupraTentorialVolNotVentVox  \\\n",
       "0         9.996803e+05              9.777723e+05                     973296.0   \n",
       "1         1.138365e+06              1.094811e+06                    1092459.0   \n",
       "2         9.026203e+05              8.838913e+05                     881444.0   \n",
       "3         1.040302e+06              1.020943e+06                    1019030.0   \n",
       "4         1.074093e+06              1.056334e+06                    1053446.0   \n",
       "...                ...                       ...                          ...   \n",
       "1906      1.123903e+06              1.030924e+06                    1027319.0   \n",
       "1907      8.354743e+05              8.187873e+05                     816977.0   \n",
       "1908               NaN                       NaN                          NaN   \n",
       "1909               NaN                       NaN                          NaN   \n",
       "1910      1.001420e+06              9.639952e+05                     962387.0   \n",
       "\n",
       "      TD_score PIGD_score  TD_PIGD_ratio  TD_PIGD_class     INFODT_  \\\n",
       "0     0.000000        0.0            NaN  Indeterminate  2011-02-01   \n",
       "1     0.272727        0.0       5.000000             TD  2011-03-01   \n",
       "2     0.181818        0.6       0.303030           PIGD  2011-03-01   \n",
       "3     0.363636        0.4       0.909091  Indeterminate  2011-04-01   \n",
       "4     0.000000        0.0            NaN  Indeterminate  2011-04-01   \n",
       "...        ...        ...            ...            ...         ...   \n",
       "1906  0.090909        0.6       0.151515           PIGD  2013-09-01   \n",
       "1907  0.000000        0.4       0.000000           PIGD  2014-07-01   \n",
       "1908  0.000000        0.0            NaN  Indeterminate  2014-03-01   \n",
       "1909  0.090909        0.0       0.000000             TD  2015-01-01   \n",
       "1910  0.000000        0.0       0.000000  Indeterminate  2015-02-01   \n",
       "\n",
       "       INFODT_bl  \n",
       "0     2011-02-01  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3     2011-04-01  \n",
       "4     2011-04-01  \n",
       "...          ...  \n",
       "1906  2013-09-01  \n",
       "1907  2014-07-01  \n",
       "1908  2014-03-01  \n",
       "1909  2015-01-01  \n",
       "1910  2015-02-01  \n",
       "\n",
       "[1911 rows x 903 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MRI = pd.read_csv(\"../Processed_Images/PPMIMERGE_20200419.csv\")\n",
    "MRI_unique = MRI.groupby('PATNO').first().reset_index()\n",
    "MRI_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "pca = pd.read_table('Data/QC/White_2.eigenvec', sep=' ', header=None)\n",
    "cnames = ['FID','IID']\n",
    "for i in range(1,11):\n",
    "    cnames.append(\"PC\"+str(i))\n",
    "pca.columns = cnames\n",
    "\n",
    "info_pca = pca.merge(ppmi_info, left_on=['FID'], right_on=['PATNO'])\n",
    "\n",
    "apprdx = list()\n",
    "\n",
    "for i in info_pca['APPRDX'].to_list():\n",
    "    if(i==1):\n",
    "        apprdx.append(\"PD\")\n",
    "    if(i==2):\n",
    "        apprdx.append(\"HC\")\n",
    "    elif(i==3):\n",
    "        apprdx.append(\"SWEDD\")\n",
    "\n",
    "info_pca['APPRDX2'] = apprdx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>TotalGrayVol</th>\n",
       "      <th>SupraTentorialVol</th>\n",
       "      <th>SupraTentorialVolNotVent</th>\n",
       "      <th>SupraTentorialVolNotVentVox</th>\n",
       "      <th>TD_score</th>\n",
       "      <th>PIGD_score</th>\n",
       "      <th>TD_PIGD_ratio</th>\n",
       "      <th>TD_PIGD_class</th>\n",
       "      <th>INFODT_</th>\n",
       "      <th>INFODT_bl</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>0.021449</td>\n",
       "      <td>-0.005234</td>\n",
       "      <td>0.009304</td>\n",
       "      <td>0.066472</td>\n",
       "      <td>0.001175</td>\n",
       "      <td>0.015063</td>\n",
       "      <td>0.052470</td>\n",
       "      <td>0.019073</td>\n",
       "      <td>...</td>\n",
       "      <td>628301.286475</td>\n",
       "      <td>9.996803e+05</td>\n",
       "      <td>9.777723e+05</td>\n",
       "      <td>973296.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-02-01</td>\n",
       "      <td>2011-02-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>0.013232</td>\n",
       "      <td>0.017714</td>\n",
       "      <td>0.011900</td>\n",
       "      <td>-0.036273</td>\n",
       "      <td>-0.034574</td>\n",
       "      <td>0.046887</td>\n",
       "      <td>0.015614</td>\n",
       "      <td>0.039156</td>\n",
       "      <td>...</td>\n",
       "      <td>668025.901257</td>\n",
       "      <td>1.138365e+06</td>\n",
       "      <td>1.094811e+06</td>\n",
       "      <td>1092459.0</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>TD</td>\n",
       "      <td>2011-03-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>-0.081716</td>\n",
       "      <td>-0.031366</td>\n",
       "      <td>-0.019860</td>\n",
       "      <td>0.000616</td>\n",
       "      <td>0.027651</td>\n",
       "      <td>-0.006289</td>\n",
       "      <td>-0.000023</td>\n",
       "      <td>-0.011180</td>\n",
       "      <td>...</td>\n",
       "      <td>585538.400359</td>\n",
       "      <td>9.026203e+05</td>\n",
       "      <td>8.838913e+05</td>\n",
       "      <td>881444.0</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2011-03-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>0.022711</td>\n",
       "      <td>0.021896</td>\n",
       "      <td>-0.009466</td>\n",
       "      <td>-0.014623</td>\n",
       "      <td>-0.035391</td>\n",
       "      <td>0.006098</td>\n",
       "      <td>0.114600</td>\n",
       "      <td>0.010290</td>\n",
       "      <td>...</td>\n",
       "      <td>640446.950904</td>\n",
       "      <td>1.074093e+06</td>\n",
       "      <td>1.056334e+06</td>\n",
       "      <td>1053446.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>2011-04-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>0.008816</td>\n",
       "      <td>0.057796</td>\n",
       "      <td>0.087630</td>\n",
       "      <td>0.031654</td>\n",
       "      <td>0.002797</td>\n",
       "      <td>0.044745</td>\n",
       "      <td>0.020734</td>\n",
       "      <td>0.000451</td>\n",
       "      <td>...</td>\n",
       "      <td>691787.053845</td>\n",
       "      <td>1.120597e+06</td>\n",
       "      <td>1.091968e+06</td>\n",
       "      <td>1089715.0</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.454545</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>2011-04-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>0.007829</td>\n",
       "      <td>0.031864</td>\n",
       "      <td>0.030776</td>\n",
       "      <td>0.007838</td>\n",
       "      <td>0.077760</td>\n",
       "      <td>-0.058105</td>\n",
       "      <td>-0.008439</td>\n",
       "      <td>-0.074725</td>\n",
       "      <td>...</td>\n",
       "      <td>559773.127620</td>\n",
       "      <td>1.166641e+06</td>\n",
       "      <td>1.118808e+06</td>\n",
       "      <td>1111258.0</td>\n",
       "      <td>1.090909</td>\n",
       "      <td>0.2</td>\n",
       "      <td>5.454545</td>\n",
       "      <td>TD</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>-0.016362</td>\n",
       "      <td>0.026210</td>\n",
       "      <td>-0.005509</td>\n",
       "      <td>-0.027007</td>\n",
       "      <td>-0.079430</td>\n",
       "      <td>-0.044746</td>\n",
       "      <td>0.009454</td>\n",
       "      <td>0.006037</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.136364</td>\n",
       "      <td>TD</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>-0.005867</td>\n",
       "      <td>-0.004077</td>\n",
       "      <td>-0.053631</td>\n",
       "      <td>0.035073</td>\n",
       "      <td>-0.001406</td>\n",
       "      <td>-0.092460</td>\n",
       "      <td>0.058593</td>\n",
       "      <td>0.027439</td>\n",
       "      <td>...</td>\n",
       "      <td>695806.656485</td>\n",
       "      <td>1.145848e+06</td>\n",
       "      <td>1.115972e+06</td>\n",
       "      <td>1113066.0</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>0.2</td>\n",
       "      <td>2.727273</td>\n",
       "      <td>TD</td>\n",
       "      <td>2013-03-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.036528</td>\n",
       "      <td>-0.014558</td>\n",
       "      <td>-0.060270</td>\n",
       "      <td>-0.033671</td>\n",
       "      <td>0.042000</td>\n",
       "      <td>-0.106515</td>\n",
       "      <td>0.008027</td>\n",
       "      <td>-0.108608</td>\n",
       "      <td>...</td>\n",
       "      <td>695968.316289</td>\n",
       "      <td>1.078311e+06</td>\n",
       "      <td>1.053845e+06</td>\n",
       "      <td>1051472.0</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2013-03-01</td>\n",
       "      <td>2013-03-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>0.039298</td>\n",
       "      <td>-0.099175</td>\n",
       "      <td>-0.063026</td>\n",
       "      <td>-0.022460</td>\n",
       "      <td>0.009492</td>\n",
       "      <td>-0.034970</td>\n",
       "      <td>-0.015220</td>\n",
       "      <td>-0.053593</td>\n",
       "      <td>...</td>\n",
       "      <td>608575.445209</td>\n",
       "      <td>9.160329e+05</td>\n",
       "      <td>8.770189e+05</td>\n",
       "      <td>876096.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>2013-04-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>475 rows × 945 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID       PC1       PC2       PC3       PC4       PC5       PC6  \\\n",
       "0    3000  3000  0.021449 -0.005234  0.009304  0.066472  0.001175  0.015063   \n",
       "1    3001  3001  0.013232  0.017714  0.011900 -0.036273 -0.034574  0.046887   \n",
       "2    3002  3002 -0.081716 -0.031366 -0.019860  0.000616  0.027651 -0.006289   \n",
       "3    3004  3004  0.022711  0.021896 -0.009466 -0.014623 -0.035391  0.006098   \n",
       "4    3006  3006  0.008816  0.057796  0.087630  0.031654  0.002797  0.044745   \n",
       "..    ...   ...       ...       ...       ...       ...       ...       ...   \n",
       "470  4124  4124  0.007829  0.031864  0.030776  0.007838  0.077760 -0.058105   \n",
       "471  4125  4125 -0.016362  0.026210 -0.005509 -0.027007 -0.079430 -0.044746   \n",
       "472  4126  4126 -0.005867 -0.004077 -0.053631  0.035073 -0.001406 -0.092460   \n",
       "473  4136  4136  0.036528 -0.014558 -0.060270 -0.033671  0.042000 -0.106515   \n",
       "474  4139  4139  0.039298 -0.099175 -0.063026 -0.022460  0.009492 -0.034970   \n",
       "\n",
       "          PC7       PC8  ...   TotalGrayVol  SupraTentorialVol  \\\n",
       "0    0.052470  0.019073  ...  628301.286475       9.996803e+05   \n",
       "1    0.015614  0.039156  ...  668025.901257       1.138365e+06   \n",
       "2   -0.000023 -0.011180  ...  585538.400359       9.026203e+05   \n",
       "3    0.114600  0.010290  ...  640446.950904       1.074093e+06   \n",
       "4    0.020734  0.000451  ...  691787.053845       1.120597e+06   \n",
       "..        ...       ...  ...            ...                ...   \n",
       "470 -0.008439 -0.074725  ...  559773.127620       1.166641e+06   \n",
       "471  0.009454  0.006037  ...            NaN                NaN   \n",
       "472  0.058593  0.027439  ...  695806.656485       1.145848e+06   \n",
       "473  0.008027 -0.108608  ...  695968.316289       1.078311e+06   \n",
       "474 -0.015220 -0.053593  ...  608575.445209       9.160329e+05   \n",
       "\n",
       "     SupraTentorialVolNotVent SupraTentorialVolNotVentVox  TD_score  \\\n",
       "0                9.777723e+05                    973296.0  0.000000   \n",
       "1                1.094811e+06                   1092459.0  0.272727   \n",
       "2                8.838913e+05                    881444.0  0.181818   \n",
       "3                1.056334e+06                   1053446.0  0.000000   \n",
       "4                1.091968e+06                   1089715.0  0.090909   \n",
       "..                        ...                         ...       ...   \n",
       "470              1.118808e+06                   1111258.0  1.090909   \n",
       "471                       NaN                         NaN  1.000000   \n",
       "472              1.115972e+06                   1113066.0  0.545455   \n",
       "473              1.053845e+06                   1051472.0  0.181818   \n",
       "474              8.770189e+05                    876096.0  0.000000   \n",
       "\n",
       "    PIGD_score TD_PIGD_ratio  TD_PIGD_class     INFODT_   INFODT_bl  \n",
       "0          0.0           NaN  Indeterminate  2011-02-01  2011-02-01  \n",
       "1          0.0      5.000000             TD  2011-03-01         NaN  \n",
       "2          0.6      0.303030           PIGD  2011-03-01         NaN  \n",
       "3          0.0           NaN  Indeterminate  2011-04-01  2011-04-01  \n",
       "4          0.2      0.454545           PIGD  2011-04-01  2011-04-01  \n",
       "..         ...           ...            ...         ...         ...  \n",
       "470        0.2      5.454545             TD  2013-04-01         NaN  \n",
       "471        0.0      1.136364             TD  2013-04-01         NaN  \n",
       "472        0.2      2.727273             TD  2013-03-01         NaN  \n",
       "473        0.6      0.303030           PIGD  2013-03-01  2013-03-01  \n",
       "474        0.0           NaN  Indeterminate  2013-04-01  2013-04-01  \n",
       "\n",
       "[475 rows x 945 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_pheno = info_pca.merge(MRI_unique, left_on=['FID'], right_on=['PATNO'])\n",
    "info_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/guglielmo/opt/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4153: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  downcast=downcast,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>APPRDX2</th>\n",
       "      <th>rh_parahippocampal_area</th>\n",
       "      <th>lh_parahippocampal_area</th>\n",
       "      <th>rh_parahippocampal_volume</th>\n",
       "      <th>lh_parahippocampal_volume</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>HC</td>\n",
       "      <td>723</td>\n",
       "      <td>680</td>\n",
       "      <td>2392</td>\n",
       "      <td>2063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>PD</td>\n",
       "      <td>721</td>\n",
       "      <td>746</td>\n",
       "      <td>2053</td>\n",
       "      <td>2129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>PD</td>\n",
       "      <td>606</td>\n",
       "      <td>503</td>\n",
       "      <td>2233</td>\n",
       "      <td>1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>HC</td>\n",
       "      <td>662</td>\n",
       "      <td>696</td>\n",
       "      <td>2127</td>\n",
       "      <td>2205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>PD</td>\n",
       "      <td>693</td>\n",
       "      <td>837</td>\n",
       "      <td>2623</td>\n",
       "      <td>3134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>PD</td>\n",
       "      <td>821</td>\n",
       "      <td>819</td>\n",
       "      <td>1582</td>\n",
       "      <td>2079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>PD</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>PD</td>\n",
       "      <td>715</td>\n",
       "      <td>789</td>\n",
       "      <td>2091</td>\n",
       "      <td>2298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>PD</td>\n",
       "      <td>843</td>\n",
       "      <td>795</td>\n",
       "      <td>2446</td>\n",
       "      <td>2321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>HC</td>\n",
       "      <td>574</td>\n",
       "      <td>591</td>\n",
       "      <td>1920</td>\n",
       "      <td>2009</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>475 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID APPRDX2 rh_parahippocampal_area lh_parahippocampal_area  \\\n",
       "0    3000  3000      HC                     723                     680   \n",
       "1    3001  3001      PD                     721                     746   \n",
       "2    3002  3002      PD                     606                     503   \n",
       "3    3004  3004      HC                     662                     696   \n",
       "4    3006  3006      PD                     693                     837   \n",
       "..    ...   ...     ...                     ...                     ...   \n",
       "470  4124  4124      PD                     821                     819   \n",
       "471  4125  4125      PD                      NA                      NA   \n",
       "472  4126  4126      PD                     715                     789   \n",
       "473  4136  4136      PD                     843                     795   \n",
       "474  4139  4139      HC                     574                     591   \n",
       "\n",
       "    rh_parahippocampal_volume lh_parahippocampal_volume  \n",
       "0                        2392                      2063  \n",
       "1                        2053                      2129  \n",
       "2                        2233                      1612  \n",
       "3                        2127                      2205  \n",
       "4                        2623                      3134  \n",
       "..                        ...                       ...  \n",
       "470                      1582                      2079  \n",
       "471                        NA                        NA  \n",
       "472                      2091                      2298  \n",
       "473                      2446                      2321  \n",
       "474                      1920                      2009  \n",
       "\n",
       "[475 rows x 7 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno = info_pheno[[\"FID\",\"IID\",\"APPRDX2\",\"rh_parahippocampal_area\",\"lh_parahippocampal_area\",\"rh_parahippocampal_volume\",\"lh_parahippocampal_volume\"]]\n",
    "pheno.fillna(\"NA\",inplace=True)\n",
    "pheno.to_csv('GWAS2/MRI/Phenotype_With_SWEDD.txt',sep=\"\\t\", index=False)\n",
    "pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "age = list()\n",
    "for i in range(len(info_pheno)):\n",
    "    mm,yy = info_pheno.loc[i,\"SITE_APRV\"].split(\"/\")\n",
    "    a = int(yy) - int(info_pheno.loc[i,\"BIRTHDT_x\"])\n",
    "    age.append(a)\n",
    "\n",
    "info_pheno[\"AGE\"] = age   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/guglielmo/opt/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4153: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  downcast=downcast,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>eTIV</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>1.56977e+06</td>\n",
       "      <td>70</td>\n",
       "      <td>0.021449</td>\n",
       "      <td>-0.005234</td>\n",
       "      <td>0.009304</td>\n",
       "      <td>0.066472</td>\n",
       "      <td>0.001175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>1.71515e+06</td>\n",
       "      <td>65</td>\n",
       "      <td>0.013232</td>\n",
       "      <td>0.017714</td>\n",
       "      <td>0.011900</td>\n",
       "      <td>-0.036273</td>\n",
       "      <td>-0.034574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>1.35646e+06</td>\n",
       "      <td>68</td>\n",
       "      <td>-0.081716</td>\n",
       "      <td>-0.031366</td>\n",
       "      <td>-0.019860</td>\n",
       "      <td>0.000616</td>\n",
       "      <td>0.027651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>1.61768e+06</td>\n",
       "      <td>60</td>\n",
       "      <td>0.022711</td>\n",
       "      <td>0.021896</td>\n",
       "      <td>-0.009466</td>\n",
       "      <td>-0.014623</td>\n",
       "      <td>-0.035391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>1.6439e+06</td>\n",
       "      <td>58</td>\n",
       "      <td>0.008816</td>\n",
       "      <td>0.057796</td>\n",
       "      <td>0.087630</td>\n",
       "      <td>0.031654</td>\n",
       "      <td>0.002797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>1.55545e+06</td>\n",
       "      <td>71</td>\n",
       "      <td>0.007829</td>\n",
       "      <td>0.031864</td>\n",
       "      <td>0.030776</td>\n",
       "      <td>0.007838</td>\n",
       "      <td>0.077760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>NA</td>\n",
       "      <td>64</td>\n",
       "      <td>-0.016362</td>\n",
       "      <td>0.026210</td>\n",
       "      <td>-0.005509</td>\n",
       "      <td>-0.027007</td>\n",
       "      <td>-0.079430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>1.8287e+06</td>\n",
       "      <td>56</td>\n",
       "      <td>-0.005867</td>\n",
       "      <td>-0.004077</td>\n",
       "      <td>-0.053631</td>\n",
       "      <td>0.035073</td>\n",
       "      <td>-0.001406</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>1.65036e+06</td>\n",
       "      <td>56</td>\n",
       "      <td>0.036528</td>\n",
       "      <td>-0.014558</td>\n",
       "      <td>-0.060270</td>\n",
       "      <td>-0.033671</td>\n",
       "      <td>0.042000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>1.54333e+06</td>\n",
       "      <td>81</td>\n",
       "      <td>0.039298</td>\n",
       "      <td>-0.099175</td>\n",
       "      <td>-0.063026</td>\n",
       "      <td>-0.022460</td>\n",
       "      <td>0.009492</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>475 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID         eTIV  AGE       PC1       PC2       PC3       PC4  \\\n",
       "0    3000  3000  1.56977e+06   70  0.021449 -0.005234  0.009304  0.066472   \n",
       "1    3001  3001  1.71515e+06   65  0.013232  0.017714  0.011900 -0.036273   \n",
       "2    3002  3002  1.35646e+06   68 -0.081716 -0.031366 -0.019860  0.000616   \n",
       "3    3004  3004  1.61768e+06   60  0.022711  0.021896 -0.009466 -0.014623   \n",
       "4    3006  3006   1.6439e+06   58  0.008816  0.057796  0.087630  0.031654   \n",
       "..    ...   ...          ...  ...       ...       ...       ...       ...   \n",
       "470  4124  4124  1.55545e+06   71  0.007829  0.031864  0.030776  0.007838   \n",
       "471  4125  4125           NA   64 -0.016362  0.026210 -0.005509 -0.027007   \n",
       "472  4126  4126   1.8287e+06   56 -0.005867 -0.004077 -0.053631  0.035073   \n",
       "473  4136  4136  1.65036e+06   56  0.036528 -0.014558 -0.060270 -0.033671   \n",
       "474  4139  4139  1.54333e+06   81  0.039298 -0.099175 -0.063026 -0.022460   \n",
       "\n",
       "          PC5  \n",
       "0    0.001175  \n",
       "1   -0.034574  \n",
       "2    0.027651  \n",
       "3   -0.035391  \n",
       "4    0.002797  \n",
       "..        ...  \n",
       "470  0.077760  \n",
       "471 -0.079430  \n",
       "472 -0.001406  \n",
       "473  0.042000  \n",
       "474  0.009492  \n",
       "\n",
       "[475 rows x 9 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = info_pheno[[\"FID\",\"IID\",\"eTIV\",\"AGE\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\"]]\n",
    "cov.fillna(\"NA\",inplace=True)\n",
    "cov.to_csv('GWAS2/MRI/Covariate_With_SWEDD.txt',sep=\"\\t\", index=False)\n",
    "cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rh_parahippocampal_area',\n",
       " 'lh_parahippocampal_area',\n",
       " 'rh_parahippocampal_volume',\n",
       " 'lh_parahippocampal_volume']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col = list(pheno.columns.values)[3:7] \n",
    "col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/rh_parahippocampal_area_With_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/MRI/Covariate_With_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/rh_parahippocampal_area_With_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_With_SWEDD.txt\n",
      "  --pheno-name rh_parahippocampal_area\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "435 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "40 people had missing value(s).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/rh_parahippocampal_area_With_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/rh_parahippocampal_area_With_SWEDD.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/lh_parahippocampal_area_With_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/MRI/Covariate_With_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/lh_parahippocampal_area_With_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_With_SWEDD.txt\n",
      "  --pheno-name lh_parahippocampal_area\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "435 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "40 people had missing value(s).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/lh_parahippocampal_area_With_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.00594.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/lh_parahippocampal_area_With_SWEDD.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/rh_parahippocampal_volume_With_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/MRI/Covariate_With_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/rh_parahippocampal_volume_With_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_With_SWEDD.txt\n",
      "  --pheno-name rh_parahippocampal_volume\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "435 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "40 people had missing value(s).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/rh_parahippocampal_volume_With_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/rh_parahippocampal_volume_With_SWEDD.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/lh_parahippocampal_volume_With_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2\n",
      "  --covar GWAS2/MRI/Covariate_With_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/lh_parahippocampal_volume_With_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_With_SWEDD.txt\n",
      "  --pheno-name lh_parahippocampal_volume\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "475 people (312 males, 163 females) loaded from .fam.\n",
      "435 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "40 people had missing value(s).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996613.\n",
      "128957 variants and 475 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/lh_parahippocampal_volume_With_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/lh_parahippocampal_volume_With_SWEDD.assoc.linear.adjusted .\n"
     ]
    }
   ],
   "source": [
    "for i in col:\n",
    "    !plink --bfile DATA/QC/White_2 --pheno GWAS2/MRI/Phenotype_With_SWEDD.txt --covar GWAS2/MRI/Covariate_With_SWEDD.txt --pheno-name {i} --covar-name eTIV,AGE,PC1-PC5 --linear sex --adjust --out GWAS2/MRI/{i+\"_With_SWEDD\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in col:\n",
    "    gwas = pd.read_table(\"GWAS2/MRI/\"+i+\"_With_SWEDD.assoc.linear\",sep=\"\\s+\")\n",
    "    gwas = gwas[gwas.TEST==\"ADD\"]\n",
    "    gwas.to_csv('GWAS2/MRI/GWAS_MRI_'+i+'_With_SWEDD.csv',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MRI GWAS WITHOUTH SWEDD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to Data/QC/White_2_no_swedd.log.\n",
      "Options in effect:\n",
      "  --bfile Data/QC/White_2_no_swedd\n",
      "  --out Data/QC/White_2_no_swedd\n",
      "  --pca 10\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "Using up to 11 threads (change this with --threads).\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to Data/QC/White_2_no_swedd.eigenval and\n",
      "Data/QC/White_2_no_swedd.eigenvec .\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile Data/QC/White_2_no_swedd --pca 10 --out Data/QC/White_2_no_swedd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>SupraTentorialVol</th>\n",
       "      <th>SupraTentorialVolNotVent</th>\n",
       "      <th>SupraTentorialVolNotVentVox</th>\n",
       "      <th>TD_score</th>\n",
       "      <th>PIGD_score</th>\n",
       "      <th>TD_PIGD_ratio</th>\n",
       "      <th>TD_PIGD_class</th>\n",
       "      <th>INFODT_</th>\n",
       "      <th>INFODT_bl</th>\n",
       "      <th>AGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>0.026252</td>\n",
       "      <td>0.000831</td>\n",
       "      <td>0.034813</td>\n",
       "      <td>0.066445</td>\n",
       "      <td>0.005376</td>\n",
       "      <td>-0.039601</td>\n",
       "      <td>0.039440</td>\n",
       "      <td>0.011572</td>\n",
       "      <td>...</td>\n",
       "      <td>9.996803e+05</td>\n",
       "      <td>9.777723e+05</td>\n",
       "      <td>973296.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-02-01</td>\n",
       "      <td>2011-02-01</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>0.017288</td>\n",
       "      <td>0.016918</td>\n",
       "      <td>0.000809</td>\n",
       "      <td>-0.046732</td>\n",
       "      <td>0.036114</td>\n",
       "      <td>-0.066129</td>\n",
       "      <td>0.008962</td>\n",
       "      <td>-0.059123</td>\n",
       "      <td>...</td>\n",
       "      <td>1.138365e+06</td>\n",
       "      <td>1.094811e+06</td>\n",
       "      <td>1092459.0</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>TD</td>\n",
       "      <td>2011-03-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>-0.083952</td>\n",
       "      <td>-0.036270</td>\n",
       "      <td>-0.023529</td>\n",
       "      <td>0.008239</td>\n",
       "      <td>-0.028558</td>\n",
       "      <td>0.007373</td>\n",
       "      <td>-0.004357</td>\n",
       "      <td>0.005666</td>\n",
       "      <td>...</td>\n",
       "      <td>9.026203e+05</td>\n",
       "      <td>8.838913e+05</td>\n",
       "      <td>881444.0</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2011-03-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>0.024155</td>\n",
       "      <td>0.033869</td>\n",
       "      <td>-0.003613</td>\n",
       "      <td>-0.008359</td>\n",
       "      <td>0.043702</td>\n",
       "      <td>-0.057734</td>\n",
       "      <td>0.070331</td>\n",
       "      <td>0.013879</td>\n",
       "      <td>...</td>\n",
       "      <td>1.074093e+06</td>\n",
       "      <td>1.056334e+06</td>\n",
       "      <td>1053446.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>0.008186</td>\n",
       "      <td>0.057054</td>\n",
       "      <td>0.101822</td>\n",
       "      <td>0.001634</td>\n",
       "      <td>-0.006341</td>\n",
       "      <td>-0.047876</td>\n",
       "      <td>-0.016801</td>\n",
       "      <td>-0.033359</td>\n",
       "      <td>...</td>\n",
       "      <td>1.120597e+06</td>\n",
       "      <td>1.091968e+06</td>\n",
       "      <td>1089715.0</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.454545</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>2011-04-01</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.030614</td>\n",
       "      <td>0.040836</td>\n",
       "      <td>-0.011779</td>\n",
       "      <td>-0.077894</td>\n",
       "      <td>0.065532</td>\n",
       "      <td>0.009551</td>\n",
       "      <td>0.081294</td>\n",
       "      <td>...</td>\n",
       "      <td>1.166641e+06</td>\n",
       "      <td>1.118808e+06</td>\n",
       "      <td>1111258.0</td>\n",
       "      <td>1.090909</td>\n",
       "      <td>0.2</td>\n",
       "      <td>5.454545</td>\n",
       "      <td>TD</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>-0.020655</td>\n",
       "      <td>0.029306</td>\n",
       "      <td>-0.017984</td>\n",
       "      <td>-0.021871</td>\n",
       "      <td>0.082911</td>\n",
       "      <td>0.017845</td>\n",
       "      <td>0.023150</td>\n",
       "      <td>0.058820</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.136364</td>\n",
       "      <td>TD</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>-0.008676</td>\n",
       "      <td>0.002931</td>\n",
       "      <td>-0.037925</td>\n",
       "      <td>0.059620</td>\n",
       "      <td>0.013533</td>\n",
       "      <td>0.038492</td>\n",
       "      <td>0.110887</td>\n",
       "      <td>0.040737</td>\n",
       "      <td>...</td>\n",
       "      <td>1.145848e+06</td>\n",
       "      <td>1.115972e+06</td>\n",
       "      <td>1113066.0</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>0.2</td>\n",
       "      <td>2.727273</td>\n",
       "      <td>TD</td>\n",
       "      <td>2013-03-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>0.033337</td>\n",
       "      <td>-0.014685</td>\n",
       "      <td>-0.064613</td>\n",
       "      <td>-0.017093</td>\n",
       "      <td>-0.033616</td>\n",
       "      <td>0.121496</td>\n",
       "      <td>0.029545</td>\n",
       "      <td>0.085788</td>\n",
       "      <td>...</td>\n",
       "      <td>1.078311e+06</td>\n",
       "      <td>1.053845e+06</td>\n",
       "      <td>1051472.0</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.303030</td>\n",
       "      <td>PIGD</td>\n",
       "      <td>2013-03-01</td>\n",
       "      <td>2013-03-01</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>0.049703</td>\n",
       "      <td>-0.100721</td>\n",
       "      <td>-0.075755</td>\n",
       "      <td>-0.010019</td>\n",
       "      <td>-0.007243</td>\n",
       "      <td>0.050604</td>\n",
       "      <td>0.005441</td>\n",
       "      <td>0.075253</td>\n",
       "      <td>...</td>\n",
       "      <td>9.160329e+05</td>\n",
       "      <td>8.770189e+05</td>\n",
       "      <td>876096.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>2013-04-01</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 946 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID       PC1       PC2       PC3       PC4       PC5       PC6  \\\n",
       "0    3000  3000  0.026252  0.000831  0.034813  0.066445  0.005376 -0.039601   \n",
       "1    3001  3001  0.017288  0.016918  0.000809 -0.046732  0.036114 -0.066129   \n",
       "2    3002  3002 -0.083952 -0.036270 -0.023529  0.008239 -0.028558  0.007373   \n",
       "3    3004  3004  0.024155  0.033869 -0.003613 -0.008359  0.043702 -0.057734   \n",
       "4    3006  3006  0.008186  0.057054  0.101822  0.001634 -0.006341 -0.047876   \n",
       "..    ...   ...       ...       ...       ...       ...       ...       ...   \n",
       "428  4124  4124  0.008458  0.030614  0.040836 -0.011779 -0.077894  0.065532   \n",
       "429  4125  4125 -0.020655  0.029306 -0.017984 -0.021871  0.082911  0.017845   \n",
       "430  4126  4126 -0.008676  0.002931 -0.037925  0.059620  0.013533  0.038492   \n",
       "431  4136  4136  0.033337 -0.014685 -0.064613 -0.017093 -0.033616  0.121496   \n",
       "432  4139  4139  0.049703 -0.100721 -0.075755 -0.010019 -0.007243  0.050604   \n",
       "\n",
       "          PC7       PC8  ...  SupraTentorialVol  SupraTentorialVolNotVent  \\\n",
       "0    0.039440  0.011572  ...       9.996803e+05              9.777723e+05   \n",
       "1    0.008962 -0.059123  ...       1.138365e+06              1.094811e+06   \n",
       "2   -0.004357  0.005666  ...       9.026203e+05              8.838913e+05   \n",
       "3    0.070331  0.013879  ...       1.074093e+06              1.056334e+06   \n",
       "4   -0.016801 -0.033359  ...       1.120597e+06              1.091968e+06   \n",
       "..        ...       ...  ...                ...                       ...   \n",
       "428  0.009551  0.081294  ...       1.166641e+06              1.118808e+06   \n",
       "429  0.023150  0.058820  ...                NaN                       NaN   \n",
       "430  0.110887  0.040737  ...       1.145848e+06              1.115972e+06   \n",
       "431  0.029545  0.085788  ...       1.078311e+06              1.053845e+06   \n",
       "432  0.005441  0.075253  ...       9.160329e+05              8.770189e+05   \n",
       "\n",
       "     SupraTentorialVolNotVentVox  TD_score  PIGD_score TD_PIGD_ratio  \\\n",
       "0                       973296.0  0.000000         0.0           NaN   \n",
       "1                      1092459.0  0.272727         0.0      5.000000   \n",
       "2                       881444.0  0.181818         0.6      0.303030   \n",
       "3                      1053446.0  0.000000         0.0           NaN   \n",
       "4                      1089715.0  0.090909         0.2      0.454545   \n",
       "..                           ...       ...         ...           ...   \n",
       "428                    1111258.0  1.090909         0.2      5.454545   \n",
       "429                          NaN  1.000000         0.0      1.136364   \n",
       "430                    1113066.0  0.545455         0.2      2.727273   \n",
       "431                    1051472.0  0.181818         0.6      0.303030   \n",
       "432                     876096.0  0.000000         0.0           NaN   \n",
       "\n",
       "     TD_PIGD_class     INFODT_   INFODT_bl  AGE  \n",
       "0    Indeterminate  2011-02-01  2011-02-01   70  \n",
       "1               TD  2011-03-01         NaN   65  \n",
       "2             PIGD  2011-03-01         NaN   68  \n",
       "3    Indeterminate  2011-04-01  2011-04-01   60  \n",
       "4             PIGD  2011-04-01  2011-04-01   58  \n",
       "..             ...         ...         ...  ...  \n",
       "428             TD  2013-04-01         NaN   71  \n",
       "429             TD  2013-04-01         NaN   64  \n",
       "430             TD  2013-03-01         NaN   56  \n",
       "431           PIGD  2013-03-01  2013-03-01   56  \n",
       "432  Indeterminate  2013-04-01  2013-04-01   81  \n",
       "\n",
       "[433 rows x 946 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pca = pd.read_table('Data/QC/White_2_no_swedd.eigenvec', sep=' ', header=None)\n",
    "cnames = ['FID','IID']\n",
    "for i in range(1,11):\n",
    "    cnames.append(\"PC\"+str(i))\n",
    "pca.columns = cnames\n",
    "\n",
    "info_pca = pca.merge(ppmi_info, left_on=['FID'], right_on=['PATNO'])\n",
    "\n",
    "apprdx = list()\n",
    "\n",
    "for i in info_pca['APPRDX'].to_list():\n",
    "    if(i==1):\n",
    "        apprdx.append(\"PD\")\n",
    "    if(i==2):\n",
    "        apprdx.append(\"HC\")\n",
    "    elif(i==3):\n",
    "        apprdx.append(\"SWEDD\")\n",
    "\n",
    "info_pca['APPRDX2'] = apprdx\n",
    "\n",
    "info_pheno = info_pca.merge(MRI_unique, left_on=['FID'], right_on=['PATNO'])\n",
    "age = list()\n",
    "for i in range(len(info_pheno)):\n",
    "    mm,yy = info_pheno.loc[i,\"SITE_APRV\"].split(\"/\")\n",
    "    a = int(yy) - int(info_pheno.loc[i,\"BIRTHDT_x\"])\n",
    "    age.append(a)\n",
    "\n",
    "info_pheno[\"AGE\"] = age   \n",
    "info_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/guglielmo/opt/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4153: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  downcast=downcast,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>APPRDX2</th>\n",
       "      <th>rh_parahippocampal_area</th>\n",
       "      <th>lh_parahippocampal_area</th>\n",
       "      <th>rh_parahippocampal_volume</th>\n",
       "      <th>lh_parahippocampal_volume</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>HC</td>\n",
       "      <td>723</td>\n",
       "      <td>680</td>\n",
       "      <td>2392</td>\n",
       "      <td>2063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>PD</td>\n",
       "      <td>721</td>\n",
       "      <td>746</td>\n",
       "      <td>2053</td>\n",
       "      <td>2129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>PD</td>\n",
       "      <td>606</td>\n",
       "      <td>503</td>\n",
       "      <td>2233</td>\n",
       "      <td>1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>HC</td>\n",
       "      <td>662</td>\n",
       "      <td>696</td>\n",
       "      <td>2127</td>\n",
       "      <td>2205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>PD</td>\n",
       "      <td>693</td>\n",
       "      <td>837</td>\n",
       "      <td>2623</td>\n",
       "      <td>3134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>PD</td>\n",
       "      <td>821</td>\n",
       "      <td>819</td>\n",
       "      <td>1582</td>\n",
       "      <td>2079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>PD</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>PD</td>\n",
       "      <td>715</td>\n",
       "      <td>789</td>\n",
       "      <td>2091</td>\n",
       "      <td>2298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>PD</td>\n",
       "      <td>843</td>\n",
       "      <td>795</td>\n",
       "      <td>2446</td>\n",
       "      <td>2321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>HC</td>\n",
       "      <td>574</td>\n",
       "      <td>591</td>\n",
       "      <td>1920</td>\n",
       "      <td>2009</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID APPRDX2 rh_parahippocampal_area lh_parahippocampal_area  \\\n",
       "0    3000  3000      HC                     723                     680   \n",
       "1    3001  3001      PD                     721                     746   \n",
       "2    3002  3002      PD                     606                     503   \n",
       "3    3004  3004      HC                     662                     696   \n",
       "4    3006  3006      PD                     693                     837   \n",
       "..    ...   ...     ...                     ...                     ...   \n",
       "428  4124  4124      PD                     821                     819   \n",
       "429  4125  4125      PD                      NA                      NA   \n",
       "430  4126  4126      PD                     715                     789   \n",
       "431  4136  4136      PD                     843                     795   \n",
       "432  4139  4139      HC                     574                     591   \n",
       "\n",
       "    rh_parahippocampal_volume lh_parahippocampal_volume  \n",
       "0                        2392                      2063  \n",
       "1                        2053                      2129  \n",
       "2                        2233                      1612  \n",
       "3                        2127                      2205  \n",
       "4                        2623                      3134  \n",
       "..                        ...                       ...  \n",
       "428                      1582                      2079  \n",
       "429                        NA                        NA  \n",
       "430                      2091                      2298  \n",
       "431                      2446                      2321  \n",
       "432                      1920                      2009  \n",
       "\n",
       "[433 rows x 7 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno = info_pheno[[\"FID\",\"IID\",\"APPRDX2\",\"rh_parahippocampal_area\",\"lh_parahippocampal_area\",\"rh_parahippocampal_volume\",\"lh_parahippocampal_volume\"]]\n",
    "pheno.fillna(\"NA\",inplace=True)\n",
    "pheno.to_csv('GWAS2/MRI/Phenotype_Without_SWEDD.txt',sep=\"\\t\", index=False)\n",
    "pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/guglielmo/opt/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4153: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  downcast=downcast,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>eTIV</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3000</td>\n",
       "      <td>3000</td>\n",
       "      <td>1.56977e+06</td>\n",
       "      <td>70</td>\n",
       "      <td>0.026252</td>\n",
       "      <td>0.000831</td>\n",
       "      <td>0.034813</td>\n",
       "      <td>0.066445</td>\n",
       "      <td>0.005376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3001</td>\n",
       "      <td>3001</td>\n",
       "      <td>1.71515e+06</td>\n",
       "      <td>65</td>\n",
       "      <td>0.017288</td>\n",
       "      <td>0.016918</td>\n",
       "      <td>0.000809</td>\n",
       "      <td>-0.046732</td>\n",
       "      <td>0.036114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3002</td>\n",
       "      <td>3002</td>\n",
       "      <td>1.35646e+06</td>\n",
       "      <td>68</td>\n",
       "      <td>-0.083952</td>\n",
       "      <td>-0.036270</td>\n",
       "      <td>-0.023529</td>\n",
       "      <td>0.008239</td>\n",
       "      <td>-0.028558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3004</td>\n",
       "      <td>3004</td>\n",
       "      <td>1.61768e+06</td>\n",
       "      <td>60</td>\n",
       "      <td>0.024155</td>\n",
       "      <td>0.033869</td>\n",
       "      <td>-0.003613</td>\n",
       "      <td>-0.008359</td>\n",
       "      <td>0.043702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3006</td>\n",
       "      <td>3006</td>\n",
       "      <td>1.6439e+06</td>\n",
       "      <td>58</td>\n",
       "      <td>0.008186</td>\n",
       "      <td>0.057054</td>\n",
       "      <td>0.101822</td>\n",
       "      <td>0.001634</td>\n",
       "      <td>-0.006341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>4124</td>\n",
       "      <td>4124</td>\n",
       "      <td>1.55545e+06</td>\n",
       "      <td>71</td>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.030614</td>\n",
       "      <td>0.040836</td>\n",
       "      <td>-0.011779</td>\n",
       "      <td>-0.077894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>4125</td>\n",
       "      <td>4125</td>\n",
       "      <td>NA</td>\n",
       "      <td>64</td>\n",
       "      <td>-0.020655</td>\n",
       "      <td>0.029306</td>\n",
       "      <td>-0.017984</td>\n",
       "      <td>-0.021871</td>\n",
       "      <td>0.082911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>4126</td>\n",
       "      <td>4126</td>\n",
       "      <td>1.8287e+06</td>\n",
       "      <td>56</td>\n",
       "      <td>-0.008676</td>\n",
       "      <td>0.002931</td>\n",
       "      <td>-0.037925</td>\n",
       "      <td>0.059620</td>\n",
       "      <td>0.013533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>4136</td>\n",
       "      <td>4136</td>\n",
       "      <td>1.65036e+06</td>\n",
       "      <td>56</td>\n",
       "      <td>0.033337</td>\n",
       "      <td>-0.014685</td>\n",
       "      <td>-0.064613</td>\n",
       "      <td>-0.017093</td>\n",
       "      <td>-0.033616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>1.54333e+06</td>\n",
       "      <td>81</td>\n",
       "      <td>0.049703</td>\n",
       "      <td>-0.100721</td>\n",
       "      <td>-0.075755</td>\n",
       "      <td>-0.010019</td>\n",
       "      <td>-0.007243</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID   IID         eTIV  AGE       PC1       PC2       PC3       PC4  \\\n",
       "0    3000  3000  1.56977e+06   70  0.026252  0.000831  0.034813  0.066445   \n",
       "1    3001  3001  1.71515e+06   65  0.017288  0.016918  0.000809 -0.046732   \n",
       "2    3002  3002  1.35646e+06   68 -0.083952 -0.036270 -0.023529  0.008239   \n",
       "3    3004  3004  1.61768e+06   60  0.024155  0.033869 -0.003613 -0.008359   \n",
       "4    3006  3006   1.6439e+06   58  0.008186  0.057054  0.101822  0.001634   \n",
       "..    ...   ...          ...  ...       ...       ...       ...       ...   \n",
       "428  4124  4124  1.55545e+06   71  0.008458  0.030614  0.040836 -0.011779   \n",
       "429  4125  4125           NA   64 -0.020655  0.029306 -0.017984 -0.021871   \n",
       "430  4126  4126   1.8287e+06   56 -0.008676  0.002931 -0.037925  0.059620   \n",
       "431  4136  4136  1.65036e+06   56  0.033337 -0.014685 -0.064613 -0.017093   \n",
       "432  4139  4139  1.54333e+06   81  0.049703 -0.100721 -0.075755 -0.010019   \n",
       "\n",
       "          PC5  \n",
       "0    0.005376  \n",
       "1    0.036114  \n",
       "2   -0.028558  \n",
       "3    0.043702  \n",
       "4   -0.006341  \n",
       "..        ...  \n",
       "428 -0.077894  \n",
       "429  0.082911  \n",
       "430  0.013533  \n",
       "431 -0.033616  \n",
       "432 -0.007243  \n",
       "\n",
       "[433 rows x 9 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = info_pheno[[\"FID\",\"IID\",\"eTIV\",\"AGE\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\"]]\n",
    "cov.fillna(\"NA\",inplace=True)\n",
    "cov.to_csv('GWAS2/MRI/Covariate_Without_SWEDD.txt',sep=\"\\t\", index=False)\n",
    "cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/rh_parahippocampal_area_Without_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/MRI/Covariate_Without_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/rh_parahippocampal_area_Without_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_Without_SWEDD.txt\n",
      "  --pheno-name rh_parahippocampal_area\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "398 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "35 people had missing value(s).\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/rh_parahippocampal_area_Without_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/rh_parahippocampal_area_Without_SWEDD.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/lh_parahippocampal_area_Without_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/MRI/Covariate_Without_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/lh_parahippocampal_area_Without_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_Without_SWEDD.txt\n",
      "  --pheno-name lh_parahippocampal_area\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "398 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "35 people had missing value(s).\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/lh_parahippocampal_area_Without_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/lh_parahippocampal_area_Without_SWEDD.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/rh_parahippocampal_volume_Without_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/MRI/Covariate_Without_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/rh_parahippocampal_volume_Without_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_Without_SWEDD.txt\n",
      "  --pheno-name rh_parahippocampal_volume\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "398 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "35 people had missing value(s).\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/rh_parahippocampal_volume_Without_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/rh_parahippocampal_volume_Without_SWEDD.assoc.linear.adjusted .\n",
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to GWAS2/MRI/lh_parahippocampal_volume_Without_SWEDD.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile DATA/QC/White_2_no_swedd\n",
      "  --covar GWAS2/MRI/Covariate_Without_SWEDD.txt\n",
      "  --covar-name eTIV,AGE,PC1-PC5\n",
      "  --linear sex\n",
      "  --out GWAS2/MRI/lh_parahippocampal_volume_Without_SWEDD\n",
      "  --pheno GWAS2/MRI/Phenotype_Without_SWEDD.txt\n",
      "  --pheno-name lh_parahippocampal_volume\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "128957 variants loaded from .bim file.\n",
      "433 people (286 males, 147 females) loaded from .fam.\n",
      "398 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 7 covariates loaded.\n",
      "35 people had missing value(s).\n",
      "Before main variant filters, 433 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996625.\n",
      "128957 variants and 433 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "GWAS2/MRI/lh_parahippocampal_volume_Without_SWEDD.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128957 variants) written to\n",
      "GWAS2/MRI/lh_parahippocampal_volume_Without_SWEDD.assoc.linear.adjusted .\n"
     ]
    }
   ],
   "source": [
    "for i in col:\n",
    "    !plink --bfile DATA/QC/White_2_no_swedd --pheno GWAS2/MRI/Phenotype_Without_SWEDD.txt --covar GWAS2/MRI/Covariate_Without_SWEDD.txt --pheno-name {i} --covar-name eTIV,AGE,PC1-PC5 --linear sex --adjust --out GWAS2/MRI/{i+\"_Without_SWEDD\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in col:\n",
    "    gwas = pd.read_table(\"GWAS2/MRI/\"+i+\"_Without_SWEDD.assoc.linear\",sep=\"\\s+\")\n",
    "    gwas = gwas[gwas.TEST==\"ADD\"]\n",
    "    gwas.to_csv('GWAS2/MRI/GWAS_MRI_'+i+'_Without_SWEDD.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
