{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, sys\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import math\n",
    "import matplotlib.pyplot as plt\n",
    "from scipy.stats import norm\n",
    "from matplotlib_venn import venn3\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_path='../PPMI_data/1_Immunochip_SNP_Data'\n",
    "neuro_path='../PPMI_data/2_NeuroX_SNP_Data_for_Original_Cohort'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "set1_bim_fn = os.path.join(immuno_path , 'IMMUNO.bim')\n",
    "set2_bim_fn = os.path.join(neuro_path ,  'NEUROX.bim')\n",
    "\n",
    "set1_bim = pd.read_table(set1_bim_fn, header=None)\n",
    "set2_bim = pd.read_table(set2_bim_fn, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>898835</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1108138</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1110294</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1125105</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1130298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5\n",
       "0  1  imm_1_898835  0   898835  0  A\n",
       "1  1  vh_1_1108138  0  1108138  T  C\n",
       "2  1  vh_1_1110294  0  1110294  A  G\n",
       "3  1     rs9729550  0  1125105  C  A\n",
       "4  1     rs1815606  0  1130298  T  G"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set1_bim.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2  3  4  5\n",
       "0  1     NeuroX_PARK7_Pro158del  0  0  0  I\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0  0  0  I\n",
       "2  1   NeuroX_PINK1_534_535insQ  0  0  0  D\n",
       "3  1      NeuroX_PINK1_Asp525fs  0  0  0  D\n",
       "4  1      NeuroX_PINK1_Cys549fs  0  0  0  I"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set2_bim.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Immuno is on hg18 genome assemply while neurox snps data is map on hg19 genome assemply, we need to map immuno to hg19 as neurox."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      0        3        3             1\n",
      "0  chr1   898835   898836  imm_1_898835\n",
      "1  chr1  1108138  1108139  vh_1_1108138\n",
      "2  chr1  1110294  1110295  vh_1_1110294\n",
      "3  chr1  1125105  1125106     rs9729550\n",
      "4  chr1  1130298  1130299     rs1815606\n",
      "            0          3          3           1\n",
      "196519   chrX  154889906  154889907   rs3093525\n",
      "196520   chrX  154889917  154889918   rs3093526\n",
      "196521   chrX  154892630  154892631   rs2742301\n",
      "196522   chrX  154907376  154907377   rs2981835\n",
      "196523  chrMT       3721       3722  MitoA3721G\n"
     ]
    }
   ],
   "source": [
    "tmp_chr=set1_bim.iloc[:,0]\n",
    "tmp_pos=set1_bim.iloc[:,3]\n",
    "tmp_name=set1_bim.iloc[:,1]\n",
    "\n",
    "#append 'chr' to the chromosome number\n",
    "tmp_chr_str = [\"chr\" + str(sub) for sub in tmp_chr.values]\n",
    "\n",
    "#replace the numbers 23, 24, 25, 26 with letters, plink way to encode regions\n",
    "tmp_chr_str2 = [w.replace('23', 'X') for w in tmp_chr_str]\n",
    "tmp_chr_str3 = [w.replace('24', 'Y') for w in tmp_chr_str2]\n",
    "tmp_chr_str4 = [w.replace('25', 'X') for w in tmp_chr_str3]\n",
    "tmp_chr_str5 = [w.replace('26', 'MT') for w in tmp_chr_str4]\n",
    "\n",
    "#put they together\n",
    "bed=pd.concat([pd.DataFrame(tmp_chr_str5), tmp_pos, tmp_pos+1, tmp_name], axis=1)\n",
    "print(bed.head())\n",
    "print(bed.tail())\n",
    "bed.to_csv(os.path.join(immuno_path , 'IMMUNO_hg18.bed'), sep=\" \", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We use the liftOver tool from the UCSC genome browser (https://genome.ucsc.edu/) to get the new positions. Some SNPs will not be converted. Commnad line only for Linux users.\n",
    "Let's load the new cooridnates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_bed=pd.read_table(os.path.join(immuno_path, 'IMMUNO_hg19_ucsc.bed'), header=None)\n",
    "\n",
    "good_snps=new_bed.iloc[:,3]\n",
    "#write successful SNPs to a file\n",
    "good_snps.to_csv(os.path.join(immuno_path, 'good_snps.txt'), index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to Data/IMMUNO_hg19.log.\n",
      "Options in effect:\n",
      "  --bfile ../PPMI_data/1_Immunochip_SNP_Data/IMMUNO\n",
      "  --extract ../PPMI_data/1_Immunochip_SNP_Data/good_snps.txt\n",
      "  --make-bed\n",
      "  --out Data/IMMUNO_hg19\n",
      "  --update-map ../PPMI_data/1_Immunochip_SNP_Data/IMMUNO_hg19_ucsc.bed 2 4\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "196524 variants loaded from .bim file.\n",
      "523 people (343 males, 180 females) loaded from .fam.\n",
      "--update-map: 196520 values updated.\n",
      "Warning: Base-pair positions are now unsorted!\n",
      "--extract: 196520 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 523 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 67239 het. haploid genotypes present (see Data/IMMUNO_hg19.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.957444.\n",
      "196520 variants and 523 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to Data/IMMUNO_hg19.bed + Data/IMMUNO_hg19.bim +\n",
      "Data/IMMUNO_hg19.fam ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, 'IMMUNO')} --update-map {os.path.join(immuno_path, 'IMMUNO_hg19_ucsc.bed')} 2 4 --make-bed --extract {os.path.join(immuno_path, 'good_snps.txt')} --out Data/IMMUNO_hg19"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merging the 2 different dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno = pd.read_table(\"Data/IMMUNO_hg19.bim\", header=None)\n",
    "neurox = set2_bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1120431</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5\n",
       "0  1  imm_1_898835  0   908972  0  A\n",
       "1  1  vh_1_1108138  0  1118275  T  C\n",
       "2  1  vh_1_1110294  0  1120431  A  G\n",
       "3  1     rs9729550  0  1135242  C  A\n",
       "4  1     rs1815606  0  1140435  T  G"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267602</th>\n",
       "      <td>26</td>\n",
       "      <td>exm2216497</td>\n",
       "      <td>0</td>\n",
       "      <td>15942</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267603</th>\n",
       "      <td>26</td>\n",
       "      <td>exm2216498</td>\n",
       "      <td>0</td>\n",
       "      <td>15946</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267604</th>\n",
       "      <td>26</td>\n",
       "      <td>exm2216499</td>\n",
       "      <td>0</td>\n",
       "      <td>15951</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267605</th>\n",
       "      <td>26</td>\n",
       "      <td>exm2216500</td>\n",
       "      <td>0</td>\n",
       "      <td>15978</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267606</th>\n",
       "      <td>26</td>\n",
       "      <td>exm-rs41378955</td>\n",
       "      <td>0</td>\n",
       "      <td>16390</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267607 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         0                          1  2      3  4  5\n",
       "0        1     NeuroX_PARK7_Pro158del  0      0  0  I\n",
       "1        1  NeuroX_PINK1_23bp_del_ex7  0      0  0  I\n",
       "2        1   NeuroX_PINK1_534_535insQ  0      0  0  D\n",
       "3        1      NeuroX_PINK1_Asp525fs  0      0  0  D\n",
       "4        1      NeuroX_PINK1_Cys549fs  0      0  0  I\n",
       "...     ..                        ... ..    ... .. ..\n",
       "267602  26                 exm2216497  0  15942  0  T\n",
       "267603  26                 exm2216498  0  15946  T  C\n",
       "267604  26                 exm2216499  0  15951  G  A\n",
       "267605  26                 exm2216500  0  15978  T  C\n",
       "267606  26             exm-rs41378955  0  16390  0  0\n",
       "\n",
       "[267607 rows x 6 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>1_908972_908973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1118275_1118276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1120431</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1120431_1120432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1_1135242_1135243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1140435_1140436</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5                  6\n",
       "0  1  imm_1_898835  0   908972  0  A    1_908972_908973\n",
       "1  1  vh_1_1108138  0  1118275  T  C  1_1118275_1118276\n",
       "2  1  vh_1_1110294  0  1120431  A  G  1_1120431_1120432\n",
       "3  1     rs9729550  0  1135242  C  A  1_1135242_1135243\n",
       "4  1     rs1815606  0  1140435  T  G  1_1140435_1140436"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coords = list()\n",
    "for i in range(len(immuno)):\n",
    "    chrom = str(immuno.iloc[i,0])\n",
    "    start = str(immuno.iloc[i,3])\n",
    "    stop  = str(immuno.iloc[i,3]+1)\n",
    "    coord = '_'.join([chrom,start,stop])\n",
    "    coords.append(coord)\n",
    "\n",
    "immuno[6] = coords\n",
    "immuno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>1_0_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>1_0_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>1_0_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_762320_762321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>1_861349_861350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1_865545_865546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865584_865585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>exm46</td>\n",
       "      <td>0</td>\n",
       "      <td>865625</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865625_865626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>exm47</td>\n",
       "      <td>0</td>\n",
       "      <td>865628</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865628_865629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1</td>\n",
       "      <td>exm51</td>\n",
       "      <td>0</td>\n",
       "      <td>865662</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865662_865663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "      <td>exm53</td>\n",
       "      <td>0</td>\n",
       "      <td>865665</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865665_865666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1</td>\n",
       "      <td>exm55</td>\n",
       "      <td>0</td>\n",
       "      <td>865694</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_865694_865695</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    0                          1  2       3  4  5                6\n",
       "0   1     NeuroX_PARK7_Pro158del  0       0  0  I            1_0_1\n",
       "1   1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I            1_0_1\n",
       "2   1   NeuroX_PINK1_534_535insQ  0       0  0  D            1_0_1\n",
       "3   1      NeuroX_PINK1_Asp525fs  0       0  0  D            1_0_1\n",
       "4   1      NeuroX_PINK1_Cys549fs  0       0  0  I            1_0_1\n",
       "5   1      NeuroX_PINK1_Lys520fs  0       0  0  I            1_0_1\n",
       "6   1                 exm2268640  0  762320  T  C  1_762320_762321\n",
       "7   1                      exm41  0  861349  0  C  1_861349_861350\n",
       "8   1                 exm1916089  0  865545  0  0  1_865545_865546\n",
       "9   1                      exm44  0  865584  0  G  1_865584_865585\n",
       "10  1                      exm46  0  865625  0  G  1_865625_865626\n",
       "11  1                      exm47  0  865628  A  G  1_865628_865629\n",
       "12  1                      exm51  0  865662  0  G  1_865662_865663\n",
       "13  1                      exm53  0  865665  A  G  1_865665_865666\n",
       "14  1                      exm55  0  865694  T  C  1_865694_865695"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coords = list()\n",
    "for i in range(len(neurox)):\n",
    "    chrom = str(neurox.iloc[i,0])\n",
    "    start = str(neurox.iloc[i,3])\n",
    "    stop  = str(neurox.iloc[i,3]+1)\n",
    "    coord = '_'.join([chrom,start,stop])\n",
    "    coords.append(coord)\n",
    "\n",
    "neurox[6] = coords\n",
    "neurox.head(n=15)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Extract the snps only in immuno but not in neurox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno.to_csv('Data/IMMUNO_coords.txt', sep=\" \", index=False, header=False)\n",
    "neurox.to_csv('Data/NEUROX_coords.txt', sep=\" \", index=False, header=False)\n",
    "\n",
    "!awk 'FNR==NR { a[$NF]; next } !($NF in a)' Data/NEUROX_coords.txt Data/IMMUNO_coords.txt > Data/onlyIMMUNO.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>1_908972_908973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1118275_1118276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1_1135242_1135243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1140435_1140436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1163804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1163804_1163805</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5                  6\n",
       "0  1  imm_1_898835  0   908972  0  A    1_908972_908973\n",
       "1  1  vh_1_1108138  0  1118275  T  C  1_1118275_1118276\n",
       "2  1     rs9729550  0  1135242  C  A  1_1135242_1135243\n",
       "3  1     rs1815606  0  1140435  T  G  1_1140435_1140436\n",
       "4  1     rs7515488  0  1163804  T  C  1_1163804_1163805"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "onlyImmuno = pd.read_csv('Data/onlyIMMUNO.txt',header=None,sep=\" \")\n",
    "onlyImmuno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "onlyImmunoSnps = onlyImmuno.iloc[:,1]\n",
    "onlyImmunoSnps.to_csv('Data/OnlyImmunoSnps.txt',index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to IMMUNO_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile Data/IMMUNO_hg19\n",
      "  --extract Data/OnlyImmunoSnps.txt\n",
      "  --make-bed\n",
      "  --out IMMUNO_extracted\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "196520 variants loaded from .bim file.\n",
      "523 people (343 males, 180 females) loaded from .fam.\n",
      "--extract: 188126 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 523 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 66779 het. haploid genotypes present (see IMMUNO_extracted.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.956509.\n",
      "188126 variants and 523 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to IMMUNO_extracted.bed + IMMUNO_extracted.bim +\n",
      "IMMUNO_extracted.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile Data/IMMUNO_hg19 --extract Data/OnlyImmunoSnps.txt --make-bed --out Data/IMMUNO_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to Data/merged_PPMI.log.\n",
      "Options in effect:\n",
      "  --bfile ../PPMI_data/2_NeuroX_SNP_Data_for_Original_Cohort/NEUROX\n",
      "  --bmerge Data/IMMUNO_extracted\n",
      "  --make-bed\n",
      "  --out Data/merged_PPMI\n",
      "\n",
      "32768 MB RAM detected; reserving 16384 MB for main workspace.\n",
      "619 people loaded from\n",
      "../PPMI_data/2_NeuroX_SNP_Data_for_Original_Cohort/NEUROX.fam.\n",
      "523 people to be merged from Data/IMMUNO_extracted.fam.\n",
      "Of these, 3 are new, while 520 are present in the base dataset.\n",
      "267607 markers loaded from\n",
      "../PPMI_data/2_NeuroX_SNP_Data_for_Original_Cohort/NEUROX.bim.\n",
      "188126 markers to be merged from Data/IMMUNO_extracted.bim.\n",
      "Of these, 188126 are new, while 0 are present in the base dataset.\n",
      "Warning: Variants 'NeuroX_PINK1_23bp_del_ex7' and 'NeuroX_PARK7_Pro158del' have\n",
      "the same position.\n",
      "Warning: Variants 'NeuroX_PINK1_534_535insQ' and 'NeuroX_PINK1_23bp_del_ex7'\n",
      "have the same position.\n",
      "Warning: Variants 'NeuroX_PINK1_Asp525fs' and 'NeuroX_PINK1_534_535insQ' have\n",
      "the same position.\n",
      "1887 more same-position warnings: see log file.\n",
      "Performing single-pass merge (622 people, 455733 variants).\n",
      "Merged fileset written to Data/merged_PPMI-merge.bed +\n",
      "Data/merged_PPMI-merge.bim + Data/merged_PPMI-merge.fam .\n",
      "455733 variants loaded from .bim file.\n",
      "622 people (411 males, 211 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 622 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 72244 het. haploid genotypes present (see Data/merged_PPMI.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.891593.\n",
      "455733 variants and 622 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to Data/merged_PPMI.bed + Data/merged_PPMI.bim +\n",
      "Data/merged_PPMI.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(neuro_path, 'NEUROX')} --bmerge Data/IMMUNO_extracted --make-bed --out Data/merged_PPMI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
