{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1- Merging genotyping datasets (*Immuno* and *NeuroX*)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import norm\n",
    "from matplotlib_venn import venn3\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "import subprocess\n",
    "import math\n",
    "import sys\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Introduction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "During this step of the preprocessing phase of the workflow, we will merge the *Immuno* and *NeuroX* genotyping datasets in a single one.\n",
    "\n",
    "Genetic variant positions in *Immuno* datasets were mapped on hg18 genome assembly, while *NeuroX* SNPs were mapped on hg19 assembly. This could cause issues and inconsistencies when merging together the two set of genomic variants. To solve this issue we must lift *Immuno* variants positions from hg18 to corresponding hg19 genomic coordinates. \n",
    "\n",
    "After *Immuno* SNP positions lifting from hg18 to hg19, we merge the two datasets. During the following steps of the workflow, will be used the merged genetic variants dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exploring *Immuno* and *NeuroX*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each dataset have 3 files:\n",
    "\n",
    "- **BIM**: contains SNPs, their genomic position (chromosome and position) and their genotype status (heterozygote or homozygous)\n",
    "    \n",
    "- **FAM**: contains information related to the subjects (sex, ID, etc.)\n",
    "    \n",
    "- **BED**: binary file representing a matrix with subjects ID on the rows and SNP IDs on the columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_path = \"../../data/genotyping/IMMUNO/\"\n",
    "neurox_path = \"../../data/genotyping/NEUROX/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explore raw genotyping data contained in the two BIMs.\n",
    "\n",
    "Let's start with *Immuno*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_bim_fn = os.path.join(immuno_path, \"IMMUNO.bim\")\n",
    "neurox_bim_fn = os.path.join(neurox_path, \"NEUROX.bim\")\n",
    "\n",
    "immuno_bim = pd.read_csv(immuno_bim_fn, sep=\"\\t\", header=None)\n",
    "neurox_bim = pd.read_csv(neurox_bim_fn, sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>898835</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1108138</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1110294</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1125105</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1130298</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1153667</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11260562</td>\n",
       "      <td>0</td>\n",
       "      <td>1155173</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6697886</td>\n",
       "      <td>0</td>\n",
       "      <td>1163474</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1_1168711</td>\n",
       "      <td>0</td>\n",
       "      <td>1168711</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6603785</td>\n",
       "      <td>0</td>\n",
       "      <td>1176365</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5\n",
       "0  1  imm_1_898835  0   898835  0  A\n",
       "1  1  vh_1_1108138  0  1108138  T  C\n",
       "2  1  vh_1_1110294  0  1110294  A  G\n",
       "3  1     rs9729550  0  1125105  C  A\n",
       "4  1     rs1815606  0  1130298  T  G\n",
       "5  1     rs7515488  0  1153667  T  C\n",
       "6  1    rs11260562  0  1155173  A  G\n",
       "7  1     rs6697886  0  1163474  A  G\n",
       "8  1     1_1168711  0  1168711  A  G\n",
       "9  1     rs6603785  0  1176365  T  A"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And let's look at *NeuroX*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2       3  4  5\n",
       "0  1     NeuroX_PARK7_Pro158del  0       0  0  I\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I\n",
       "2  1   NeuroX_PINK1_534_535insQ  0       0  0  D\n",
       "3  1      NeuroX_PINK1_Asp525fs  0       0  0  D\n",
       "4  1      NeuroX_PINK1_Cys549fs  0       0  0  I\n",
       "5  1      NeuroX_PINK1_Lys520fs  0       0  0  I\n",
       "6  1                 exm2268640  0  762320  T  C\n",
       "7  1                      exm41  0  861349  0  C\n",
       "8  1                 exm1916089  0  865545  0  0\n",
       "9  1                      exm44  0  865584  0  G"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Lifting *Immuno* SNP positions to hg19 genome assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As mentioned above, the *Immuno* genetic variants positions are mapped on hg18, while those of *NeuroX* are mapped on hg19 genome assembly. To solve this problem we lift *Immuno* SNP positions to the corresponding hg19 genomic locations. \n",
    "\n",
    "To lift SNP positions we use the web-based tool liftOver at the UCSC web-server (https://genome.ucsc.edu/cgi-bin/hgLiftOver).\n",
    "\n",
    "Since liftOver accepts as input UCSC BED files and not BIMs, we first convert *Immuno* BIM file in the corresponding UCSC BED file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      0        3        3             1\n",
      "0  chr1   898835   898836  imm_1_898835\n",
      "1  chr1  1108138  1108139  vh_1_1108138\n",
      "2  chr1  1110294  1110295  vh_1_1110294\n",
      "3  chr1  1125105  1125106     rs9729550\n",
      "4  chr1  1130298  1130299     rs1815606\n",
      "            0          3          3           1\n",
      "196519   chrX  154889906  154889907   rs3093525\n",
      "196520   chrX  154889917  154889918   rs3093526\n",
      "196521   chrX  154892630  154892631   rs2742301\n",
      "196522   chrX  154907376  154907377   rs2981835\n",
      "196523  chrMT       3721       3722  MitoA3721G\n"
     ]
    }
   ],
   "source": [
    "# minimal UCSC BED file requires\n",
    "#\n",
    "#CHROM    START    STOP    FEATURE_ID \n",
    "\n",
    "tmp_chr = immuno_bim.iloc[:,0]  # chromosomes\n",
    "tmp_pos = immuno_bim.iloc[:,3]  # SNP positions\n",
    "tmp_name = immuno_bim.iloc[:,1] # SNP IDs\n",
    "\n",
    "# append 'chr' in front of chromosome numbers (required by liftOver)\n",
    "tmp_chr_str = [''.join([\"chr\", str(c)]) for c in tmp_chr.values]\n",
    "\n",
    "# replace chr23, chr24, chr25 and chr26 with chrX, chrY, chrX and chrMT\n",
    "tmp_chr_str2 = [c.replace('23', 'X') for c in tmp_chr_str]\n",
    "tmp_chr_str3 = [c.replace('24', 'Y') for c in tmp_chr_str2]\n",
    "tmp_chr_str4 = [c.replace('25', 'X') for c in tmp_chr_str3]\n",
    "tmp_chr_str5 = [c.replace('26', 'MT') for c in tmp_chr_str4]\n",
    "\n",
    "# write the resulting UCSC BED file\n",
    "bed = pd.concat(\n",
    "    [pd.DataFrame(tmp_chr_str5), tmp_pos, tmp_pos+1, tmp_name],\n",
    "    axis=1\n",
    ")\n",
    "print(bed.head(n=5))\n",
    "print(bed.tail(n=5))\n",
    "bed.to_csv(\n",
    "    os.path.join(immuno_path, \"IMMUNO_tolift.bed\"), sep=\"\\t\",\n",
    "    index=False, header=False\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can map *Immuno* positions to hg19 genome assembly using liftOver (https://genome.ucsc.edu/cgi-bin/hgLiftOver)\n",
    "\n",
    "LiftOver mapped succesfully 196,520 of IMMUNO SNPs of the original 196,524 genetic variants.\n",
    "\n",
    "The four unmapped SNPs were:\n",
    "- imm_3_50875337\n",
    "- rs1574660\n",
    "- rs1131012\n",
    "- MitoA3721G\n",
    "\n",
    "Before proceeding, we rename the resulting UCSC BED file with the lifted SNPs as ```IMMUNO_lifted.bed```."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_lifted = pd.read_csv(os.path.join(immuno_path, \"IMMUNO_lifted.bed\"), sep=\"\\t\", header=None)\n",
    "\n",
    "# we take SNPs whose conversion to hg19 positions was succesful\n",
    "good_snps = bed_lifted.iloc[:,3]\n",
    "good_snps.to_csv(\n",
    "    os.path.join(immuno_path, \"good_snps.txt\"),\n",
    "    index=False, header=False\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We keep the mapped SNPs and we create the corresponding BIM, FAM and BED files (named ```IMMUNO_hg19```)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../../data/genotyping/IMMUNO/IMMUNO_hg19.log.\n",
      "Options in effect:\n",
      "  --bfile ../../data/genotyping/IMMUNO/IMMUNO\n",
      "  --extract ../../data/genotyping/IMMUNO/good_snps.txt\n",
      "  --make-bed\n",
      "  --out ../../data/genotyping/IMMUNO/IMMUNO_hg19\n",
      "  --update-map ../../data/genotyping/IMMUNO/IMMUNO_lifted.bed 2 4\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "196524 variants loaded from .bim file.\n",
      "523 people (343 males, 180 females) loaded from .fam.\n",
      "--update-map: 196520 values updated.\n",
      "Warning: Base-pair positions are now unsorted!\n",
      "--extract: 196520 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 523 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 67239 het. haploid genotypes present (see\n",
      "../../data/genotyping/IMMUNO/IMMUNO_hg19.hh ); many commands treat these as\n",
      "missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.957444.\n",
      "196520 variants and 523 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ../../data/genotyping/IMMUNO/IMMUNO_hg19.bed +\n",
      "../../data/genotyping/IMMUNO/IMMUNO_hg19.bim +\n",
      "../../data/genotyping/IMMUNO/IMMUNO_hg19.fam ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, \"IMMUNO\")} --update-map {os.path.join(immuno_path, \"IMMUNO_lifted.bed\")} 2 4 --make-bed --extract {os.path.join(immuno_path, \"good_snps.txt\")} --out {os.path.join(immuno_path, \"IMMUNO_hg19\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merging *Immuno* and *NeuroX*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since both *Immuno* and *NeuroX* SNP positions are now mapped on the same genome assembly (hg19) we can merge the two datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1120431</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1163804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11260562</td>\n",
       "      <td>0</td>\n",
       "      <td>1165310</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6697886</td>\n",
       "      <td>0</td>\n",
       "      <td>1173611</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1_1168711</td>\n",
       "      <td>0</td>\n",
       "      <td>1178848</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6603785</td>\n",
       "      <td>0</td>\n",
       "      <td>1186502</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5\n",
       "0  1  imm_1_898835  0   908972  0  A\n",
       "1  1  vh_1_1108138  0  1118275  T  C\n",
       "2  1  vh_1_1110294  0  1120431  A  G\n",
       "3  1     rs9729550  0  1135242  C  A\n",
       "4  1     rs1815606  0  1140435  T  G\n",
       "5  1     rs7515488  0  1163804  T  C\n",
       "6  1    rs11260562  0  1165310  A  G\n",
       "7  1     rs6697886  0  1173611  A  G\n",
       "8  1     1_1168711  0  1178848  A  G\n",
       "9  1     rs6603785  0  1186502  T  A"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_bim = pd.read_csv(os.path.join(immuno_path, \"IMMUNO_hg19.bim\"), sep=\"\\t\", header=None)  # BIM with SNPs mapped on hg19\n",
    "immuno_bim.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2       3  4  5\n",
       "0  1     NeuroX_PARK7_Pro158del  0       0  0  I\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I\n",
       "2  1   NeuroX_PINK1_534_535insQ  0       0  0  D\n",
       "3  1      NeuroX_PINK1_Asp525fs  0       0  0  D\n",
       "4  1      NeuroX_PINK1_Cys549fs  0       0  0  I\n",
       "5  1      NeuroX_PINK1_Lys520fs  0       0  0  I\n",
       "6  1                 exm2268640  0  762320  T  C\n",
       "7  1                      exm41  0  861349  0  C\n",
       "8  1                 exm1916089  0  865545  0  0\n",
       "9  1                      exm44  0  865584  0  G"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_bim.head(n=10)  # look again at NeuroX"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before merging *Immuno* and *NeuroX*, we keep only those subjects with genotyping data available in both the considered datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "immuno_fam_fn = os.path.join(immuno_path, \"IMMUNO_hg19.fam\")\n",
    "neurox_fam_fn = os.path.join(neurox_path, \"NEUROX.fam\")\n",
    "\n",
    "immuno_fam = pd.read_csv(immuno_fam_fn, sep=\" \", header=None)\n",
    "neurox_fam = pd.read_csv(neurox_fam_fn, sep=\" \", header=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's explore the two FAM files content.\n",
    "\n",
    "Let's begin with *Immuno*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3400</td>\n",
       "      <td>3400</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3401</td>\n",
       "      <td>3401</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3402</td>\n",
       "      <td>3402</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3403</td>\n",
       "      <td>3403</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3404</td>\n",
       "      <td>3404</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0     1  2  3  4  5\n",
       "0  3400  3400  0  0  2 -9\n",
       "1  3401  3401  0  0  2 -9\n",
       "2  3402  3402  0  0  1 -9\n",
       "3  3403  3403  0  0  1 -9\n",
       "4  3404  3404  0  0  2 -9"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_fam.head(n=5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And *NeuroX*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3527</td>\n",
       "      <td>3527</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3274</td>\n",
       "      <td>3274</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3220</td>\n",
       "      <td>3220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3467</td>\n",
       "      <td>3467</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3171</td>\n",
       "      <td>3171</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0     1  2  3  4  5\n",
       "0  3527  3527  0  0  1 -9\n",
       "1  3274  3274  0  0  1 -9\n",
       "2  3220  3220  0  0  2 -9\n",
       "3  3467  3467  0  0  1 -9\n",
       "4  3171  3171  0  0  1 -9"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_fam.head(n=5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And we retrieve the FAMID and IIDs of those subjects with genotyping data in both *Immuno* and *NeuroX*. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "immuno_subj = immuno_fam.iloc[:,1].tolist()\n",
    "neurox_subj = neurox_fam.iloc[:,1].tolist()\n",
    "\n",
    "# get the subjects appearing in both IMMUNO and NEUROX data\n",
    "common_subj = set(immuno_subj).intersection(set(neurox_subj))\n",
    "\n",
    "# write the common subjects to a file\n",
    "with open(\"../../data/genotyping/common_subj.txt\", mode=\"w+\") as outfile:\n",
    "    for s in common_subj:\n",
    "        outfile.write(''.join([str(s), \" \", str(s), \"\\n\"]))\n",
    "        \n",
    "common_subj_fn = \"../../data/genotyping/common_subj.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Originally, in *Immuno* there are 523 subjects, while in *NeuroX* there are 619 subjects. \n",
    "\n",
    "The subjects whose genotyping data are available in both the datasets are 520. \n",
    "\n",
    "Before merging *Immuno* and *NeuroX*, we keep data belonging only to the subjects shared by the two considered datsets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../../data/genotyping/IMMUNO/IMMUNO_common.log.\n",
      "Options in effect:\n",
      "  --bfile ../../data/genotyping/IMMUNO/IMMUNO_hg19\n",
      "  --keep ../../data/genotyping/common_subj.txt\n",
      "  --make-bed\n",
      "  --out ../../data/genotyping/IMMUNO/IMMUNO_common\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "196520 variants loaded from .bim file.\n",
      "523 people (343 males, 180 females) loaded from .fam.\n",
      "--keep: 520 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 66842 het. haploid genotypes present (see\n",
      "../../data/genotyping/IMMUNO/IMMUNO_common.hh ); many commands treat these as\n",
      "missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.957428.\n",
      "196520 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ../../data/genotyping/IMMUNO/IMMUNO_common.bed +\n",
      "../../data/genotyping/IMMUNO/IMMUNO_common.bim +\n",
      "../../data/genotyping/IMMUNO/IMMUNO_common.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, \"IMMUNO_hg19\")} --keep {common_subj_fn} --make-bed --out {os.path.join(immuno_path, \"IMMUNO_common\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../../data/genotyping/NEUROX/NEUROX_common.log.\n",
      "Options in effect:\n",
      "  --bfile ../../data/genotyping/NEUROX/NEUROX\n",
      "  --keep ../../data/genotyping/common_subj.txt\n",
      "  --make-bed\n",
      "  --out ../../data/genotyping/NEUROX/NEUROX_common\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "267607 variants loaded from .bim file.\n",
      "619 people (409 males, 210 females) loaded from .fam.\n",
      "--keep: 520 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 4672 het. haploid genotypes present (see\n",
      "../../data/genotyping/NEUROX/NEUROX_common.hh ); many commands treat these as\n",
      "missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.957466.\n",
      "267607 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ../../data/genotyping/NEUROX/NEUROX_common.bed +\n",
      "../../data/genotyping/NEUROX/NEUROX_common.bim +\n",
      "../../data/genotyping/NEUROX/NEUROX_common.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(neurox_path, \"NEUROX\")} --keep {common_subj_fn} --make-bed --out {os.path.join(neurox_path, \"NEUROX_common\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PPMI documentation tells us that the genotyping was made using custom arrays. Thus, the SNP names do not always are identified with standard rsIDs. Moreover, we want to keep as much SNPs as possible from the two datasets.\n",
    "\n",
    "To get the SNPs which are uniquely in *Immuno*, we need to work on their genomic coordinates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#function to identify snps by chrom_pos_pos+1\n",
    "def rename_SNP(snpname, chrom, pos, a1, a2):\n",
    "    if chrom == 0 or pos == 0:\n",
    "        return snpname  # manage unmapped SNPs on neurox\n",
    "    return \"_\".join([str(chrom), str(pos), str(pos+1), str(a1), str(a2)])\n",
    "\n",
    "# the BIM are those containing the common subjects data\n",
    "immuno_bim = pd.read_csv(\n",
    "    os.path.join(immuno_path, \"IMMUNO_common.bim\"), sep=\"\\t\",\n",
    "    header=None\n",
    ")\n",
    "neurox_bim = pd.read_csv(\n",
    "    os.path.join(neurox_path, \"NEUROX_common.bim\"), sep=\"\\t\",\n",
    "    header=None\n",
    ")\n",
    "\n",
    "renamedsnps_immuno = immuno_bim.apply(lambda row : rename_SNP(row[1], row[0], row[3], row[4], row[5]), axis=1)\n",
    "immuno_coords = immuno_bim\n",
    "immuno_coords[6] = renamedsnps_immuno\n",
    "\n",
    "renamedsnps_neurox = neurox_bim.apply(lambda row : rename_SNP(row[1], row[0], row[3], row[4], row[5]), axis=1)\n",
    "neurox_coords = neurox_bim\n",
    "neurox_coords[6] = renamedsnps_neurox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>imm_1_898835</td>\n",
       "      <td>0</td>\n",
       "      <td>908972</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>1_908972_908973_0_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1108138</td>\n",
       "      <td>0</td>\n",
       "      <td>1118275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1118275_1118276_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>vh_1_1110294</td>\n",
       "      <td>0</td>\n",
       "      <td>1120431</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1120431_1120432_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs9729550</td>\n",
       "      <td>0</td>\n",
       "      <td>1135242</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1_1135242_1135243_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs1815606</td>\n",
       "      <td>0</td>\n",
       "      <td>1140435</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1140435_1140436_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7515488</td>\n",
       "      <td>0</td>\n",
       "      <td>1163804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_1163804_1163805_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11260562</td>\n",
       "      <td>0</td>\n",
       "      <td>1165310</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1165310_1165311_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6697886</td>\n",
       "      <td>0</td>\n",
       "      <td>1173611</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1173611_1173612_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1_1168711</td>\n",
       "      <td>0</td>\n",
       "      <td>1178848</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1_1178848_1178849_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>rs6603785</td>\n",
       "      <td>0</td>\n",
       "      <td>1186502</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1_1186502_1186503_T_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0             1  2        3  4  5                      6\n",
       "0  1  imm_1_898835  0   908972  0  A    1_908972_908973_0_A\n",
       "1  1  vh_1_1108138  0  1118275  T  C  1_1118275_1118276_T_C\n",
       "2  1  vh_1_1110294  0  1120431  A  G  1_1120431_1120432_A_G\n",
       "3  1     rs9729550  0  1135242  C  A  1_1135242_1135243_C_A\n",
       "4  1     rs1815606  0  1140435  T  G  1_1140435_1140436_T_G\n",
       "5  1     rs7515488  0  1163804  T  C  1_1163804_1163805_T_C\n",
       "6  1    rs11260562  0  1165310  A  G  1_1165310_1165311_A_G\n",
       "7  1     rs6697886  0  1173611  A  G  1_1173611_1173612_A_G\n",
       "8  1     1_1168711  0  1178848  A  G  1_1178848_1178849_A_G\n",
       "9  1     rs6603785  0  1186502  T  A  1_1186502_1186503_T_A"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immuno_coords.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>NeuroX_PARK7_Pro158del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>NeuroX_PINK1_23bp_del_ex7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>NeuroX_PINK1_534_535insQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>D</td>\n",
       "      <td>NeuroX_PINK1_Asp525fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>NeuroX_PINK1_Cys549fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>I</td>\n",
       "      <td>NeuroX_PINK1_Lys520fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>exm2268640</td>\n",
       "      <td>0</td>\n",
       "      <td>762320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1_762320_762321_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>exm41</td>\n",
       "      <td>0</td>\n",
       "      <td>861349</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>1_861349_861350_0_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>exm1916089</td>\n",
       "      <td>0</td>\n",
       "      <td>865545</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1_865545_865546_0_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>exm44</td>\n",
       "      <td>0</td>\n",
       "      <td>865584</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>1_865584_865585_0_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0                          1  2       3  4  5                          6\n",
       "0  1     NeuroX_PARK7_Pro158del  0       0  0  I     NeuroX_PARK7_Pro158del\n",
       "1  1  NeuroX_PINK1_23bp_del_ex7  0       0  0  I  NeuroX_PINK1_23bp_del_ex7\n",
       "2  1   NeuroX_PINK1_534_535insQ  0       0  0  D   NeuroX_PINK1_534_535insQ\n",
       "3  1      NeuroX_PINK1_Asp525fs  0       0  0  D      NeuroX_PINK1_Asp525fs\n",
       "4  1      NeuroX_PINK1_Cys549fs  0       0  0  I      NeuroX_PINK1_Cys549fs\n",
       "5  1      NeuroX_PINK1_Lys520fs  0       0  0  I      NeuroX_PINK1_Lys520fs\n",
       "6  1                 exm2268640  0  762320  T  C        1_762320_762321_T_C\n",
       "7  1                      exm41  0  861349  0  C        1_861349_861350_0_C\n",
       "8  1                 exm1916089  0  865545  0  0        1_865545_865546_0_0\n",
       "9  1                      exm44  0  865584  0  G        1_865584_865585_0_G"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neurox_coords.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# we get the names of the SNPs only available in IMMUNO dataset\n",
    "# and write the SNPs uniquely appearing in IMMUNO to a TXT file\n",
    "only_immuno_snps_fn = \"../../data/genotyping/onlyIMMUNOsnps.txt\"\n",
    "neurox_coords_set = set(neurox_coords.iloc[:,6].to_list())\n",
    "immuno_coords[~immuno_coords[6].isin(neurox_coords_set)].iloc[:,1].to_csv(only_immuno_snps_fn, header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then, in *Immuno* we keep only those SNPs which are not available in *NeuroX* dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.log.\n",
      "Options in effect:\n",
      "  --bfile ../../data/genotyping/IMMUNO/IMMUNO_common\n",
      "  --extract ../../data/genotyping/onlyIMMUNOsnps.txt\n",
      "  --make-bed\n",
      "  --out ../../data/genotyping/IMMUNO/IMMUNO_uniquesnps\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "196520 variants loaded from .bim file.\n",
      "520 people (341 males, 179 females) loaded from .fam.\n",
      "--extract: 189564 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 66671 het. haploid genotypes present (see\n",
      "../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.hh ); many commands treat these\n",
      "as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.956414.\n",
      "189564 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.bed +\n",
      "../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.bim +\n",
      "../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "!plink --bfile {os.path.join(immuno_path, \"IMMUNO_common\")} --extract {only_immuno_snps_fn} --make-bed --out {os.path.join(immuno_path, \"IMMUNO_uniquesnps\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can now merge *Immuno* and *NeuroX* in a single datset.\n",
    "\n",
    "Note that are created also the corresponding BIM, FAM and BED, representing the merged data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.12 64-bit (28 Oct 2019)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../../data/genotyping/PPMI_merge.log.\n",
      "Options in effect:\n",
      "  --bfile ../../data/genotyping/NEUROX/NEUROX_common\n",
      "  --bmerge ../../data/genotyping/IMMUNO/IMMUNO_uniquesnps\n",
      "  --make-bed\n",
      "  --out ../../data/genotyping/PPMI_merge\n",
      "\n",
      "16384 MB RAM detected; reserving 8192 MB for main workspace.\n",
      "520 people loaded from ../../data/genotyping/NEUROX/NEUROX_common.fam.\n",
      "520 people to be merged from\n",
      "../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.fam.\n",
      "Of these, 0 are new, while 520 are present in the base dataset.\n",
      "267607 markers loaded from ../../data/genotyping/NEUROX/NEUROX_common.bim.\n",
      "189564 markers to be merged from\n",
      "../../data/genotyping/IMMUNO/IMMUNO_uniquesnps.bim.\n",
      "Of these, 189564 are new, while 0 are present in the base dataset.\n",
      "Warning: Variants 'NeuroX_PINK1_23bp_del_ex7' and 'NeuroX_PARK7_Pro158del' have\n",
      "the same position.\n",
      "Warning: Variants 'NeuroX_PINK1_534_535insQ' and 'NeuroX_PINK1_23bp_del_ex7'\n",
      "have the same position.\n",
      "Warning: Variants 'NeuroX_PINK1_Asp525fs' and 'NeuroX_PINK1_534_535insQ' have\n",
      "the same position.\n",
      "3325 more same-position warnings: see log file.\n",
      "Performing single-pass merge (520 people, 457171 variants).\n",
      "Merged fileset written to ../../data/genotyping/PPMI_merge-merge.bed +\n",
      "../../data/genotyping/PPMI_merge-merge.bim +\n",
      "../../data/genotyping/PPMI_merge-merge.fam .\n",
      "457171 variants loaded from .bim file.\n",
      "520 people (341 males, 179 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 520 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 71343 het. haploid genotypes present (see\n",
      "../../data/genotyping/PPMI_merge.hh ); many commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.95703.\n",
      "457171 variants and 520 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ../../data/genotyping/PPMI_merge.bed +\n",
      "../../data/genotyping/PPMI_merge.bim + ../../data/genotyping/PPMI_merge.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "merged_ppmi_fn = \"../../data/genotyping/PPMI_merge\"\n",
    "\n",
    "!plink --bfile {os.path.join(neurox_path, \"NEUROX_common\")} --bmerge {os.path.join(immuno_path, \"IMMUNO_uniquesnps\")} --make-bed --out {merged_ppmi_fn}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
